{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.lines import Line2D\n",
    "from matplotlib import patches as mpatches\n",
    "import matplotlib.ticker as tck\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import matplotlib as mpl\n",
    "from os import system\n",
    "from IPython.display import clear_output\n",
    "from scipy.stats import pearsonr\n",
    "from scipy.stats import spearmanr\n",
    "import scipy.stats as stats\n",
    "import phik \n",
    "import seaborn as sns\n",
    "from phik import resources, report\n",
    "from phik.report import plot_correlation_matrix\n",
    "from itertools import combinations\n",
    "import random\n",
    "from tabulate import tabulate\n",
    "from IPython.display import display, HTML\n",
    "import graphviz\n",
    "import networkx as nx\n",
    "import itertools\n",
    "import statsmodels.stats.multitest as multi\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "from sklearn.metrics import accuracy_score,roc_auc_score,f1_score, confusion_matrix, ConfusionMatrixDisplay,classification_report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "\n",
    "class add_path():\n",
    "    def __init__(self, path):\n",
    "        self.path = path\n",
    "\n",
    "    def __enter__(self):\n",
    "        sys.path.insert(0, self.path)\n",
    "\n",
    "    def __exit__(self, exc_type, exc_value, traceback):\n",
    "        try:\n",
    "            sys.path.remove(self.path)\n",
    "        except ValueError:\n",
    "            pass\n",
    "with add_path('/home/hshadman/.local/lib/python3.7/site-packages'):\n",
    "    matplotlib = __import__('matplotlib')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'3.5.3'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "from graphviz import Digraph\n",
    "gra = Digraph()\n",
    "gra.node('a', 'Machine Learning Errors')\n",
    "\n",
    "gra.node('b', 'RMSE')\n",
    "\n",
    "gra.node('c', 'MAE')\n",
    "gra.edges(['ab', 'ac'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 2.30.1 (20180420.1509)\n",
       " -->\n",
       "<!-- Title: %3 Pages: 1 -->\n",
       "<svg width=\"264pt\" height=\"116pt\"\n",
       " viewBox=\"0.00 0.00 264.00 116.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(1 1) rotate(0) translate(4 112)\">\n",
       "<title>%3</title>\n",
       "<polygon fill=\"white\" stroke=\"white\" points=\"-4,5 -4,-112 261,-112 261,5 -4,5\"/>\n",
       "<!-- a -->\n",
       "<g id=\"node1\" class=\"node\"><title>a</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"128\" cy=\"-90\" rx=\"128.533\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"128\" y=\"-86.3\" font-family=\"Times,serif\" font-size=\"14.00\">Machine Learning Errors</text>\n",
       "</g>\n",
       "<!-- b -->\n",
       "<g id=\"node2\" class=\"node\"><title>b</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"82\" cy=\"-18\" rx=\"39.9464\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"82\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">RMSE</text>\n",
       "</g>\n",
       "<!-- a&#45;&gt;b -->\n",
       "<g id=\"edge1\" class=\"edge\"><title>a&#45;&gt;b</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M116.865,-72.055C111.384,-63.7147 104.665,-53.4908 98.594,-44.2517\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"101.45,-42.2239 93.0328,-35.789 95.5996,-46.0682 101.45,-42.2239\"/>\n",
       "</g>\n",
       "<!-- c -->\n",
       "<g id=\"node3\" class=\"node\"><title>c</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"174\" cy=\"-18\" rx=\"33.1991\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"174\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">MAE</text>\n",
       "</g>\n",
       "<!-- a&#45;&gt;c -->\n",
       "<g id=\"edge2\" class=\"edge\"><title>a&#45;&gt;c</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M139.135,-72.055C144.697,-63.592 151.533,-53.1897 157.673,-43.8447\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"160.717,-45.5862 163.284,-35.307 154.867,-41.7419 160.717,-45.5862\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.dot.Digraph at 0x2aaabe29ff98>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "digraph {\n",
      "\ta [label=\"Machine Learning Errors\"]\n",
      "\tb [label=RMSE]\n",
      "\tc [label=MAE]\n",
      "\ta -> b\n",
      "\ta -> c\n",
      "}\n"
     ]
    }
   ],
   "source": [
    "print(gra.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 2.30.1 (20180420.1509)\n",
       " -->\n",
       "<!-- Title: %3 Pages: 1 -->\n",
       "<svg width=\"206pt\" height=\"44pt\"\n",
       " viewBox=\"0.00 0.00 206.00 44.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(1 1) rotate(0) translate(4 40)\">\n",
       "<title>%3</title>\n",
       "<polygon fill=\"white\" stroke=\"white\" points=\"-4,5 -4,-40 203,-40 203,5 -4,5\"/>\n",
       "<!-- A -->\n",
       "<g id=\"node1\" class=\"node\"><title>A</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"27\" cy=\"-18\" rx=\"27\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"27\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">A</text>\n",
       "</g>\n",
       "<!-- X -->\n",
       "<g id=\"node2\" class=\"node\"><title>X</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"99\" cy=\"-18\" rx=\"27\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"99\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">X</text>\n",
       "</g>\n",
       "<!-- A&#45;&gt;X -->\n",
       "<g id=\"edge1\" class=\"edge\"><title>A&#45;&gt;X</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M54,-18C56.6147,-18 59.2295,-18 61.8442,-18\"/>\n",
       "<polygon fill=\"black\" stroke=\"black\" points=\"61.9297,-21.5001 71.9297,-18 61.9297,-14.5001 61.9297,-21.5001\"/>\n",
       "</g>\n",
       "<!-- C -->\n",
       "<g id=\"node3\" class=\"node\"><title>C</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"171\" cy=\"-18\" rx=\"27\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"171\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">C</text>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.dot.Digraph at 0x2aabe09634e0>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gra = Digraph()\n",
    "with gra.subgraph() as i:\n",
    "\n",
    "    i.attr(rank='same')\n",
    "\n",
    "    i.node('A')\n",
    "\n",
    "    i.node('X')\n",
    "\n",
    "gra.node('C')\n",
    "gra.edges(['AX'])\n",
    "gra\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/svg+xml": [
       "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n",
       "<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\"\n",
       " \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n",
       "<!-- Generated by graphviz version 2.30.1 (20180420.1509)\n",
       " -->\n",
       "<!-- Title: %3 Pages: 1 -->\n",
       "<svg width=\"62pt\" height=\"116pt\"\n",
       " viewBox=\"0.00 0.00 62.00 116.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n",
       "<g id=\"graph0\" class=\"graph\" transform=\"scale(1 1) rotate(0) translate(4 112)\">\n",
       "<title>%3</title>\n",
       "<polygon fill=\"white\" stroke=\"white\" points=\"-4,5 -4,-112 59,-112 59,5 -4,5\"/>\n",
       "<!-- A -->\n",
       "<g id=\"node1\" class=\"node\"><title>A</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"27\" cy=\"-90\" rx=\"27\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"27\" y=\"-86.3\" font-family=\"Times,serif\" font-size=\"14.00\">A</text>\n",
       "</g>\n",
       "<!-- B -->\n",
       "<g id=\"node2\" class=\"node\"><title>B</title>\n",
       "<ellipse fill=\"none\" stroke=\"black\" cx=\"27\" cy=\"-18\" rx=\"27\" ry=\"18\"/>\n",
       "<text text-anchor=\"middle\" x=\"27\" y=\"-14.3\" font-family=\"Times,serif\" font-size=\"14.00\">B</text>\n",
       "</g>\n",
       "<!-- A&#45;&#45;B -->\n",
       "<g id=\"edge1\" class=\"edge\"><title>A&#45;&#45;B</title>\n",
       "<path fill=\"none\" stroke=\"black\" d=\"M27,-71.6966C27,-60.8463 27,-46.9167 27,-36.1043\"/>\n",
       "</g>\n",
       "</g>\n",
       "</svg>\n"
      ],
      "text/plain": [
       "<graphviz.dot.Graph at 0x2aaabe29fc50>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import graphviz\n",
    "g = graphviz.Graph()\n",
    "g.node('A')\n",
    "g.node('B')\n",
    "g.edge('A','B')\n",
    "g"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# load functions \n",
    "# load the datasets, set cutoff values, correlation method (these are done individually so the system saves them globallly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_toil_dataset = '../gtex_RSEM_gene_tpm.gz'\n",
    "correlation_method ='pearson'\n",
    "tcga_hiseq_dataset = '/home/hshadman/integrins_expression/saghar_TCGA_Breast/TCGA_BRCA_HiSeqV2' \n",
    "tcga_toil_pancanc = \"../gtex_and_tcga/tcga_RSEM_gene_tpm.gz\"\n",
    "tcga_hiseq_corresponding_sample_info = '/home/hshadman/integrins_expression/saghar_TCGA_Breast/TCGA.BRCA.sampleMap_BRCA_clinicalMatrix'\n",
    "tcga_toil_sampinfo = \"../gtex_and_tcga/TCGA_phenotype_denseDataOnlyDownload.tsv.gz\"\n",
    "tcga_toil_clincinfo = \"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\"\n",
    "opposing_corr_cutoff = 0.1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#functions for data processing\n",
    "def gtex_data_processing(gtex_toil_dataset):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']\n",
    "    gtex_breast_data_all = pd.read_csv(gtex_toil_dataset,sep='\\t')\n",
    "    gtex_breast_data_all=gtex_breast_data_all.set_index('sample').copy()\n",
    "\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    gtex_breast_data_all.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    gtex_breast_data_all=gtex_breast_data_all.loc[gtex_breast_data_all.index.isin(list(gene_map.id.values))]\n",
    "    gtex_breast_data_all=pd.concat([gtex_breast_data_all,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:gtex_breast_data_all.shape[1]]\n",
    "    gtex_breast_data_all=gtex_breast_data_all.T.astype('float').copy()\n",
    "    df_int = gtex_breast_data_all.copy()\n",
    "    #drop unwanted integrins    \n",
    "    df_int=df_int.drop([undesired_integrin for undesired_integrin in df_int.columns if undesired_integrin in unwanted_integrins],axis=1)\n",
    "    #drop duplicate gene names (if any) \n",
    "    df_int=df_int.loc[:,~df_int.columns.duplicated()]    \n",
    "    \n",
    "    gtex_toil_phenotype = pd.read_csv('../GTEX_phenotype.gz',sep='\\t')\n",
    "    #subset for breast\n",
    "    gtex_toil_phenotype = gtex_toil_phenotype[gtex_toil_phenotype['body_site_detail (SMTSD)']=='Breast - Mammary Tissue']\n",
    "    #subset for female gender\n",
    "    gtex_toil_phenotype = gtex_toil_phenotype[gtex_toil_phenotype._gender=='female']\n",
    "\n",
    "    #select breast female gtex gene expression data\n",
    "    df_int=df_int.loc[df_int.index.isin(gtex_toil_phenotype.Sample.values)].copy()\n",
    "    \n",
    "    return df_int\n",
    "\n",
    "def tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    global tcga_brca_df, tcga_brca_sample_info\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]\n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop metastatic from main sample type data frame\n",
    "    #tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Metastatic')]\n",
    " \n",
    "    #separate normal and primtum (remember primary tumor contains multiple stages)\n",
    "    tcga_brca_normal_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Solid Tissue Normal']\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Primary Tumor']\n",
    "\n",
    "    #drop index of samples not in tcga_brca_df\n",
    "    tcga_brca_normal_patients = tcga_brca_normal_patients.drop(non_match for non_match in tcga_brca_normal_patients.index.values if non_match not in tcga_brca_df.columns)\n",
    "    tcga_brca_primtum_patients = tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "    #make sure both normal and tumor patients match, and drop patients not in tcga_brca_df\n",
    "    matched_patient_index_list=[]\n",
    "    for patient in tcga_brca_primtum_patients._PATIENT.values:\n",
    "        if patient in tcga_brca_normal_patients._PATIENT.values and patient in ['-'.join(patient_id.split('-')[0:3]) for patient_id in tcga_brca_df.columns]:\n",
    "            matched_patient_index_list.append(tcga_brca_sample_info[tcga_brca_sample_info._PATIENT==patient].index.values)\n",
    "    matched_patient_list=tcga_brca_sample_info.loc[[x for l in matched_patient_index_list for x in l]].copy()\n",
    "\n",
    "\n",
    "    # matched_patient_id_list=pd.Index(tcga_brca_normal_patients['_PATIENT']).intersection(pd.Index(tcga_brca_primtum_patients['_PATIENT']))\n",
    "    # matched_patient_index_list=tcga_brca_sample_info[tcga_brca_sample_info._PATIENT.isin(matched_patient_id_list)].index.values\n",
    "    # matched_patient_list = tcga_brca_sample_info.loc[[l for l in matched_patient_index_list]].copy()\n",
    "    # #ensure samples are present in tcga_brca_df\n",
    "    # matched_patient_list=matched_patient_list.loc[matched_patient_list.index.intersection(pd.Index(tcga_brca_df.columns))]\n",
    "\n",
    "    #separate normal and primary tumor\n",
    "    normal_matched_patient_list=matched_patient_list[matched_patient_list.sample_type=='Solid Tissue Normal']\n",
    "    primtum_matched_patient_list = matched_patient_list[matched_patient_list.sample_type=='Primary Tumor']\n",
    "\n",
    "    #matched normal and primary tumor samples\n",
    "    tcga_brca_df_normal_matched=tcga_brca_df[tcga_brca_df.columns[tcga_brca_df.columns.isin(normal_matched_patient_list.index.values)]]\n",
    "    tcga_brca_df_primtum_matched=tcga_brca_df[tcga_brca_df.columns[tcga_brca_df.columns.isin(primtum_matched_patient_list.index.values)]]\n",
    "\n",
    "    #test to see if normal and primary tumor patient IDs are a match\n",
    "    for i in [patient_id.split('-')[2] for patient_id in tcga_brca_df_normal_matched.columns]:\n",
    "        if i not in [patient_id.split('-')[2] for patient_id in tcga_brca_df_primtum_matched.columns]:\n",
    "            print('trouble')\n",
    "\n",
    "    #TCGA brca normal ALL\n",
    "    tcga_brca_df_normal_all = tcga_brca_df.T.loc[tcga_brca_normal_patients.index.values].copy()\n",
    "    # dataframe with all TCGA primary tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_all= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_normal_matched':tcga_brca_df_normal_matched,\n",
    "           'tcga_brca_df_primtum_matched':tcga_brca_df_primtum_matched,\n",
    "           'tcga_brca_df_normal_all':tcga_brca_df_normal_all,\n",
    "           'tcga_brca_df_primtum_all':tcga_brca_df_primtum_all}\n",
    "\n",
    "def correlation_selected_gene(provided_dataset,interest_gene,correlation_method):\n",
    "    dataset_corr = provided_dataset.corrwith(provided_dataset[interest_gene],\n",
    "                                             method=correlation_method).drop(interest_gene)\n",
    "    dataset_corr = dataset_corr.dropna(axis=0,how='any')\n",
    "    idx = dataset_corr.index.drop_duplicates(keep=False)\n",
    "    dataset_corr = dataset_corr.loc[idx]\n",
    "    return dataset_corr\n",
    "def pairwise_pearson_correlation(provided_dataset):\n",
    "    #columns have to be variables and rows have to be observations\n",
    "    pairwise_corr=pd.DataFrame(np.corrcoef(provided_dataset,rowvar=False),\n",
    "                               columns=provided_dataset.columns,index=provided_dataset.columns)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def early_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]\n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop metastatic from main sample type data frame\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Metastatic')]\n",
    "    #drop normal samples\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Solid Tissue Normal')]    \n",
    "\n",
    "    #make sure only primary tumor sample types\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Primary Tumor']\n",
    "    \n",
    "    #stage I and II only\n",
    "    #early_stages=['Stage IIB', 'Stage IA','Stage IIA','Stage I','Stage IB','Stage II']    \n",
    "    \n",
    "    #stage I only\n",
    "    early_stages=[ 'Stage IA','Stage I','Stage IB']    \n",
    "    \n",
    "    tcga_brca_primtum_patients=tcga_brca_primtum_patients[tcga_brca_primtum_patients.ajcc_pathologic_tumor_stage.isin(early_stages)]\n",
    "    \n",
    "    #drop index of samples not in tcga_brca_df\n",
    "\n",
    "    tcga_brca_primtum_patients =tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "\n",
    "    # dataframe with all TCGA primary early stage tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_early_stage= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_primtum_early_stage':tcga_brca_df_primtum_early_stage}\n",
    "\n",
    "def late_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]    \n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop normal samples\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Solid Tissue Normal')]    \n",
    "    #make sure only primary tumor sample types\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info\n",
    "\n",
    "    late_stages=['Stage IIIA', 'Stage IV','Stage IIIC','Stage IIIB','Stage III']    \n",
    "    \n",
    "    tcga_brca_primtum_patients=tcga_brca_primtum_patients[tcga_brca_primtum_patients.ajcc_pathologic_tumor_stage.isin(late_stages)]\n",
    "    \n",
    "    #drop index of samples not in tcga_brca_df\n",
    "\n",
    "    tcga_brca_primtum_patients =tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "\n",
    "    # dataframe with all TCGA primary late stage tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_late_stage= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_primtum_late_stage':tcga_brca_df_primtum_late_stage}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#functions for counting\n",
    "def count_genes_3_no_overlap(interest_gene,correlation_method,gtex_dataset,dataset2,dataset3):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    print('no overlap between datasets')\n",
    "    dict_empty={}\n",
    "    first_dataset_name = 'gtex_all'\n",
    "    second_dataset_name= 'tcga_normal_all'\n",
    "    third_dataset_name= 'tcga_primtum_all'\n",
    "    names = [first_dataset_name,second_dataset_name,third_dataset_name]\n",
    "    j=0\n",
    "    for dataset in [gtex_dataset,dataset2,dataset3]:\n",
    "        dataset=correlation_selected_gene(dataset,interest_gene,correlation_method)\n",
    "        dict_empty[names[j]]=dataset.shape[0]\n",
    "        j+=1\n",
    "    dict_pd_df = pd.DataFrame(dict_empty,index=[f'corr_with_{interest_gene}'])\n",
    "    return display(HTML(dict_pd_df.to_html()))    \n",
    "def count_genes_same_3(interest_gene,correlation_method,gtex_dataset,dataset2,dataset3):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    dataset2=correlation_selected_gene(dataset2,interest_gene,correlation_method)\n",
    "    dataset3=correlation_selected_gene(dataset3,interest_gene,correlation_method)\n",
    "    common_gene_list= gtex_dataset.index.intersection(dataset2.index).intersection(dataset3.index)\n",
    "    global new_dataset\n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],\n",
    "                             dataset2[common_gene_list],\n",
    "                             dataset3[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_all'\n",
    "    second_dataset_name= 'tcga_normal_all'\n",
    "    third_dataset_name= 'tcga_primtum_all'\n",
    "    new_dataset.columns=[first_dataset_name,second_dataset_name,third_dataset_name]\n",
    "    print(\"1st: common genes identified for all 3 datasets\\n2nd: they were counted\")\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'corr_with_{interest_gene}_all_same_genes'],\n",
    "                             columns=[first_dataset_name,second_dataset_name,third_dataset_name])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "def count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,*processed_datasets):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    print('no overlap between datasets')\n",
    "    dict_empty={}\n",
    "    j=0\n",
    "    for dataset in processed_datasets:\n",
    "        dataset=correlation_selected_gene(dataset,interest_gene,correlation_method)\n",
    "        dict_empty[input(f'insert name for {j+1}th dataset')]=dataset[dataset>=correlation_cutoff].shape[0]\n",
    "        \n",
    "        j+=1\n",
    "    \n",
    "    dict_pd_df = pd.DataFrame(dict_empty,index=[f'HIGHLY_correlated_with_{interest_gene}'])\n",
    "    #return display(HTML(dict_pd_df.to_html()))\n",
    "    return dict_pd_df\n",
    "def cutoff_applied_togtex_2_high_both(interest_gene,correlation_method,correlation_cutoff,gtex_dataset,other_dataset):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    gtex_dataset = gtex_dataset[gtex_dataset>=correlation_cutoff]\n",
    "    dataset2=correlation_selected_gene(other_dataset,interest_gene,correlation_method)\n",
    "    common_gene_list= gtex_dataset.index.intersection(dataset2.index)\n",
    "    global new_dataset    \n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],dataset2[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_cutoffmet'\n",
    "    other_dataset_name=input('insert name for other dataset')\n",
    "    new_dataset.columns=[first_dataset_name,other_dataset_name]\n",
    "    print(\"1st: high cutoff applied to gtex dataset\\n2nd: common genes identified with other dataset\\n3rd: both datasets combined\\n4th: High cutoff applied to non-gtex dataset\")\n",
    "    \n",
    "    new_dataset = new_dataset[new_dataset[other_dataset_name]>=correlation_cutoff]\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'high_corr_with_{interest_gene}_both_same_genes'],\n",
    "                             columns=[first_dataset_name,other_dataset_name])\n",
    "    return dict_pd_df\n",
    "def cutoff_applied_togtex_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,gtex_dataset,other_dataset):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    gtex_dataset = gtex_dataset[gtex_dataset>=correlation_cutoff]\n",
    "    dataset2=correlation_selected_gene(other_dataset,interest_gene,correlation_method)\n",
    "    common_gene_list= gtex_dataset.index.intersection(dataset2.index)\n",
    "    global new_dataset    \n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],dataset2[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_cutoffmet'\n",
    "    other_dataset_name=input('insert name for other dataset')\n",
    "    new_dataset.columns=[first_dataset_name,other_dataset_name]\n",
    "    print(\"1st: high cutoff applied to gtex dataset\\n2nd: common genes identified with other dataset\\n3rd: both datasets combined\\n4th: low cutoff applied to non-gtex dataset\")\n",
    "    \n",
    "    new_dataset = new_dataset[new_dataset[other_dataset_name]<=opposing_corr_cutoff]\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'corr_with_{interest_gene}_highgtex_low_other'],\n",
    "                             columns=[first_dataset_name,other_dataset_name])\n",
    "    return dict_pd_df\n",
    "def cutoff_applied_totcga_2_high_both(interest_gene,correlation_method,correlation_cutoff,gtex_dataset,other_dataset):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    dataset2=correlation_selected_gene(other_dataset,interest_gene,correlation_method)\n",
    "    dataset2 = dataset2[dataset2>=correlation_cutoff]\n",
    "    common_gene_list= gtex_dataset.index.intersection(dataset2.index)\n",
    "    global new_dataset    \n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],dataset2[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_all'\n",
    "    other_dataset_name='tcga_primtum_all_cutoffmet'\n",
    "    new_dataset.columns=[first_dataset_name,other_dataset_name]\n",
    "    print(\"1st: high cutoff applied to tcga dataset\\n2nd: common genes identified with gtex dataset\\n3rd: both datasets combined\\n4th: High cutoff applied to gtex dataset\")\n",
    "\n",
    "    new_dataset = new_dataset[new_dataset[first_dataset_name]>=correlation_cutoff]\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'high_corr_with_{interest_gene}_both_same_genes'],\n",
    "                             columns=[first_dataset_name,other_dataset_name])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "def cutoff_applied_totcga_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,gtex_dataset,other_dataset):\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    dataset2=correlation_selected_gene(other_dataset,interest_gene,correlation_method)\n",
    "    dataset2 = dataset2[dataset2>=correlation_cutoff]\n",
    "    common_gene_list= gtex_dataset.index.intersection(dataset2.index)\n",
    "    global new_dataset    \n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],dataset2[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_all'\n",
    "    other_dataset_name='tcga_primtum_all_cutoffmet'\n",
    "    new_dataset.columns=[first_dataset_name,other_dataset_name]\n",
    "    print(\"1st: high cutoff applied to tcga dataset\\n2nd: common genes identified with gtex dataset\\n3rd: both datasets combined\\n4th: low cutoff applied to gtex dataset\")\n",
    "\n",
    "    new_dataset = new_dataset[new_dataset[first_dataset_name]<=opposing_corr_cutoff]\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'corr_with_{interest_gene}_both_same_genes'],\n",
    "                             columns=[first_dataset_name,other_dataset_name])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "\n",
    "def cutoff_applied_togtex_3_high_all(interest_gene,correlation_method,correlation_cutoff,gtex_dataset,other_dataset,\n",
    "                                    dataset3):\n",
    "#dataset3 MUST BE tcga primtum all\n",
    "#make sure to preprocess dataset\n",
    "#dictionary only takes unique values, careful about this\n",
    "    gtex_dataset=correlation_selected_gene(gtex_dataset,interest_gene,correlation_method)\n",
    "    gtex_dataset = gtex_dataset[gtex_dataset>=correlation_cutoff]\n",
    "    dataset2=correlation_selected_gene(other_dataset,interest_gene,correlation_method)\n",
    "    dataset3=correlation_selected_gene(dataset3,interest_gene,correlation_method)\n",
    "    common_gene_list= (gtex_dataset.index.intersection(dataset2.index)).intersection(dataset3.index)\n",
    "    global new_dataset    \n",
    "    new_dataset = pd.concat([gtex_dataset[common_gene_list],\n",
    "                             dataset2[common_gene_list],\n",
    "                            dataset3[common_gene_list]],axis=1)\n",
    "    first_dataset_name = 'gtex_cutoffmet'\n",
    "    other_dataset_name='tcga_normal_all'\n",
    "    dataset3_name = 'tcga_primtum_all'\n",
    "    new_dataset.columns=[first_dataset_name,other_dataset_name,dataset3_name]\n",
    "    print(\"1st: high cutoff applied to gtex dataset\\n2nd: common genes identified with other datasets\\n3rd: all datasets combined\\n4th: High cutoff applied to non-gtex datasets\")\n",
    "    \n",
    "    new_dataset = new_dataset[(new_dataset[other_dataset_name]>=correlation_cutoff) & (new_dataset[dataset3_name]>=correlation_cutoff)]\n",
    "    dict_pd_df = pd.DataFrame(data=[[new_dataset.shape[0],new_dataset.shape[0],new_dataset.shape[0]]],\n",
    "                             index=[f'high_corr_with_{interest_gene}_all_same_genes'],\n",
    "                             columns=[first_dataset_name,other_dataset_name,dataset3_name])\n",
    "    return dict_pd_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#organize as a table (original way -- before Jesse finalized the presentation but i used this for table 2)\n",
    "# careful of the #ered data (e.g. #3 is #3 in both functions but presentation might be different)\n",
    "def make_data_presentable(interest_gene,correlation_method, correlation_cutoff,opposing_corr_cutoff,gtex_gct_dataset,\n",
    "                         tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info):\n",
    "    df_int=gtex_data_processing(gtex_gct_dataset)\n",
    "    tcga_data=tcga_data_processing(tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info)\n",
    "    tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "    tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "    #1\n",
    "    count_genes_3_no_overlap(interest_gene,correlation_method,df_int,tcga_brca_df_normal_all,tcga_brca_df_primtum_all)\n",
    "    #2\n",
    "    count_genes_same_3(interest_gene,correlation_method,df_int,tcga_brca_df_normal_all,tcga_brca_df_primtum_all)\n",
    "    #3\n",
    "    print('for 1th dataset ENTER: gtex_all')\n",
    "    print('for 2th dataset ENTER: tcga_normal_all')\n",
    "    count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #4\n",
    "    #print('for 1th dataset ENTER: gtex_all')\n",
    "    #print('for 2th dataset ENTER: tcga_normal_all')    \n",
    "    #4 is not very meaningful\n",
    "    #count_high_corr_genes_with_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #5\n",
    "    print('for dataset ENTER: tcga_normal_all')    \n",
    "    cutoff_applied_togtex_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #6\n",
    "    print('for dataset ENTER: tcga_normal_all')        \n",
    "    cutoff_applied_togtex_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #7\n",
    "    print('for dataset ENTER: tcga_primtum_all')            \n",
    "    cutoff_applied_togtex_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    #8\n",
    "    print('for dataset ENTER: tcga_primtum_all')                \n",
    "    cutoff_applied_togtex_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    #9\n",
    "    cutoff_applied_totcga_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    #10\n",
    "    cutoff_applied_totcga_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#table 1\n",
    "def organize_gtex_tcga_table1(interest_gene,correlation_method, correlation_cutoff,opposing_corr_cutoff,gtex_gct_dataset,\n",
    "                         tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info):\n",
    "    df_int=gtex_data_processing(gtex_gct_dataset)\n",
    "    tcga_data=tcga_data_processing(tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info)\n",
    "    tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "    tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "    #3\n",
    "    print('for 1th dataset ENTER: gtex_all')\n",
    "    print('for 2th dataset ENTER: tcga_normal_all')\n",
    "    number_3=count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #4\n",
    "    #print('for 1th dataset ENTER: gtex_all')\n",
    "    #print('for 2th dataset ENTER: tcga_normal_all')    \n",
    "    #4 is not very meaningful\n",
    "    #count_high_corr_genes_with_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #5\n",
    "    print('for dataset ENTER: tcga_normal_all')    \n",
    "    number_5=cutoff_applied_togtex_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #7\n",
    "    print('for dataset ENTER: tcga_primtum_all')            \n",
    "    nnumber_7=cutoff_applied_togtex_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    #10\n",
    "    number_10=cutoff_applied_togtex_3_high_all(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all,\n",
    "                                    tcga_brca_df_primtum_all)\n",
    "    first_column_name='Integrin' #index\n",
    "    second_column_name='GTEX_highly_correlated'\n",
    "    third_column_name='%also_highly_correlated_in_TCGA_normal'\n",
    "    fourth_column_name='%also_highly_correlated_in_TCGA_primtum'\n",
    "    fifth_column_name='%also_highly_correlated_in_all_three'\n",
    "    #ignoring first column name. setting index as integrin name (first column name)\n",
    "    dict_pd_df = pd.DataFrame(data=[[number_3.iloc[0,0],\n",
    "                                     str(format((number_5.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_5.iloc[0,0])})',\n",
    "                                     str(format((nnumber_7.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(nnumber_7.iloc[0,0])})',\n",
    "                                     str(format((number_10.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_10.iloc[0,0])})',\n",
    "                                    ]],\n",
    "                              columns=[second_column_name,third_column_name,\n",
    "                                      fourth_column_name,fifth_column_name],\n",
    "                              index=[interest_gene])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "\n",
    "def organize_gtex_tcga_table2(interest_gene,correlation_method, correlation_cutoff,opposing_corr_cutoff,gtex_gct_dataset,\n",
    "                         tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info):\n",
    "    df_int=gtex_data_processing(gtex_gct_dataset)\n",
    "    tcga_data=tcga_data_processing(tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info)\n",
    "    tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "    tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "    #3\n",
    "    print('for 1th dataset ENTER: gtex_all')\n",
    "    print('for 2th dataset ENTER: tcga_normal_all')\n",
    "    number_3=count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #4\n",
    "    #print('for 1th dataset ENTER: gtex_all')\n",
    "    #print('for 2th dataset ENTER: tcga_normal_all')    \n",
    "    #4 is not very meaningful\n",
    "    #count_high_corr_genes_with_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #6\n",
    "    print('for dataset ENTER: tcga_normal_all')        \n",
    "    number_6=cutoff_applied_togtex_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #8\n",
    "    print('for dataset ENTER: tcga_primtum_all')                \n",
    "    number_8=cutoff_applied_togtex_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    first_column_name='Integrin' #index\n",
    "    second_column_name='GTEX_highly_correlated'\n",
    "    third_column_name='%lowly_correlated_in_TCGA_normal'\n",
    "    fourth_column_name='%lowly_correlated_in_TCGA_primtum'\n",
    "    #fifth_column_name='%also_highly_correlated_in_all_three'\n",
    "    #ignoring first column name. setting index as integrin name (first column name)\n",
    "    dict_pd_df = pd.DataFrame(data=[[number_3.iloc[0,0],\n",
    "                                     str(format((number_6.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_6.iloc[0,0])})',\n",
    "                                     str(format((number_8.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_8.iloc[0,0])})'\n",
    "                                    ]],\n",
    "                              columns=[second_column_name,third_column_name,\n",
    "                                      fourth_column_name],\n",
    "                              index=[interest_gene])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "\n",
    "#table 3\n",
    "def organize_no_overlap_table3(interest_gene,correlation_method, correlation_cutoff,opposing_corr_cutoff,gtex_gct_dataset,\n",
    "                         tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info):\n",
    "    df_int=gtex_data_processing(gtex_gct_dataset)\n",
    "    tcga_data=tcga_data_processing(tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info)\n",
    "    tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "    tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "    \n",
    "    gtex_expression_mean = df_int[interest_gene].mean()\n",
    "    gtex_expression_stdev= df_int[interest_gene].std()\n",
    "    \n",
    "    #3\n",
    "    print('for 1th dataset ENTER: gtex_all')\n",
    "    print('for 2th dataset ENTER: tcga_normal_all')\n",
    "    number_3_tcga_normal=count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #3\n",
    "    print('for 1th dataset ENTER: gtex_all')\n",
    "    print('for 2th dataset ENTER: tcga_primtum_all')\n",
    "    number_3_tcga_primtum=count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    \n",
    "\n",
    "    first_column_name='Integrin' #index\n",
    "    second_column_name='GTEX_expression_mean_&_stdev'\n",
    "    third_column_name='GTEX_highly_correlated'\n",
    "    fourth_column_name='highly_correlated_in_TCGA_normal'\n",
    "    fifth_column_name='highly_corelated_in_TCGA_primtum'\n",
    "    #ignoring first column name. setting index as integrin name (first column name)\n",
    "    dict_pd_df = pd.DataFrame(data=[[f'{format(gtex_expression_mean,\"0.2f\")} + {format(gtex_expression_stdev,\"0.2f\")}',\n",
    "                                     str(number_3_tcga_normal.iloc[0,0]),\n",
    "                                     str(number_3_tcga_normal.iloc[0,1]),\n",
    "                                     str(number_3_tcga_primtum.iloc[0,1])]],\n",
    "                              columns=[second_column_name,third_column_name,\n",
    "                                      fourth_column_name,fifth_column_name],\n",
    "                              index=[interest_gene])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "    \n",
    "def organize_tcga_gtex_table4(interest_gene,correlation_method, correlation_cutoff,opposing_corr_cutoff,gtex_gct_dataset,\n",
    "                         tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info):\n",
    "#designed for custom changes to helper functions in below cell (way down below)\n",
    "#some helper functions appear twice because too many input variables so i made changes \n",
    "#changes in same cell as table 4 function\n",
    "    df_int=gtex_data_processing(gtex_gct_dataset)\n",
    "    tcga_data=tcga_data_processing(tcga_hiseq_dataset,tcga_hiseq_corresponding_sample_info)\n",
    "    tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "    tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "    #3\n",
    "    print('its TCGA_primtum_all')\n",
    "\n",
    "    number_3=count_high_corr_genes_no_overlap(interest_gene,correlation_method,correlation_cutoff,'tcga_primtum',tcga_brca_df_primtum_all)\n",
    "    #4\n",
    "    #print('for 1th dataset ENTER: gtex_all')\n",
    "    #print('for 2th dataset ENTER: tcga_normal_all')    \n",
    "    #4 is not very meaningful\n",
    "    #count_high_corr_genes_with_overlap(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_normal_all)\n",
    "    #6\n",
    "    print('common gene list applied to all')\n",
    "    print('high tcga_primtum and low tcga normal')        \n",
    "    number_6=cutoff_applied_totcga_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,\n",
    "                                              tcga_brca_df_normal_all,tcga_brca_df_primtum_all,'tcga_normal')\n",
    "    #8\n",
    "    print('high tcga_primtum and low gtex')                \n",
    "    number_8=cutoff_applied_totcga_2_high_low(interest_gene,correlation_method,correlation_cutoff,opposing_corr_cutoff,\n",
    "                                              df_int,tcga_brca_df_primtum_all,'gtex_all')\n",
    "    #9a\n",
    "    print('high tcga primtum, high gtex')\n",
    "    number_9a = cutoff_applied_totcga_2_high_both(interest_gene,correlation_method,correlation_cutoff,df_int,tcga_brca_df_primtum_all)\n",
    "    first_column_name='Integrin' #index\n",
    "    second_column_name='TCGA_tumor_highly_correlated'\n",
    "    third_column_name='%lowly_correlated_in_TCGA_normal'\n",
    "    fourth_column_name='%lowly_correlated_in_gtex'\n",
    "    fifth_column_name = '%highly_corelated_in_gtex'\n",
    "    #fifth_column_name='%also_highly_correlated_in_all_three'\n",
    "    #ignoring first column name. setting index as integrin name (first column name)\n",
    "    dict_pd_df = pd.DataFrame(data=[[number_3.iloc[0,0],\n",
    "                                     str(format((number_6.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_6.iloc[0,0])})',\n",
    "                                     str(format((number_8.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_8.iloc[0,0])})',\n",
    "                                     str(format((number_9a.iloc[0,0]/number_3.iloc[0,0])*100,'2f'))+f'({str(number_9a.iloc[0,0])})'\n",
    "                                    ]],\n",
    "                              columns=[second_column_name,third_column_name,\n",
    "                                      fourth_column_name,fifth_column_name],\n",
    "                              index=[interest_gene])\n",
    "    return display(HTML(dict_pd_df.to_html()))\n",
    "    \n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "before dropping duplicates 19165 19165\n",
      "after dropping duplicates 19158 19158\n",
      "before dropping duplicates 19165 19165\n",
      "after dropping duplicates 19158 19158\n",
      "before dropping duplicates 19165 19165\n",
      "after dropping duplicates 19158 19158\n",
      "before dropping duplicates 19165 19165\n",
      "after dropping duplicates 19158 19158\n"
     ]
    }
   ],
   "source": [
    "#files loading\n",
    "df_int=gtex_data_processing(gtex_toil_dataset)\n",
    "tcga_data=tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)\n",
    "tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "tcga_brca_df_normal_matched=tcga_data['tcga_brca_df_normal_matched']\n",
    "tcga_brca_df_primtum_matched=tcga_data['tcga_brca_df_primtum_matched']\n",
    "tcga_brca_df_primtum_early_stage=early_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)['tcga_brca_df_primtum_early_stage']\n",
    "tcga_brca_df_primtum_late_stage=late_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)['tcga_brca_df_primtum_late_stage']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# calculate pairwise correlation PEARSON (end star protocols section)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# checking top genes most correlated with each integrin, then seeing overlap between each dataset, then comparing with other gene networking analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ITGA10', 'ITGAD', 'ITGAM', 'ITGA3', 'ITGBL1', 'ITGAE', 'ITGA2',\n",
       "       'ITGB3', 'ITGA7', 'ITGB8', 'ITGAX', 'ITGAV', 'ITGB6', 'ITGB7',\n",
       "       'ITGA5', 'ITGB4', 'ITGB2', 'ITGA6', 'ITGA2B', 'ITGB1', 'ITGAL',\n",
       "       'ITGA9', 'ITGB5', 'ITGA8', 'ITGA4', 'ITGA1', 'ITGA11'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "itg_list = df_int[df_int.columns[df_int.columns.str.contains('ITG')]].columns.values\n",
    "itg_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/public/apps/python/3.7.0/lib/python3.7/site-packages/numpy/lib/function_base.py:2559: RuntimeWarning: invalid value encountered in true_divide\n",
      "  c /= stddev[:, None]\n",
      "/public/apps/python/3.7.0/lib/python3.7/site-packages/numpy/lib/function_base.py:2560: RuntimeWarning: invalid value encountered in true_divide\n",
      "  c /= stddev[None, :]\n"
     ]
    }
   ],
   "source": [
    "#df_int is gtex dataframe. CAUTION multiple instances of dataframe variable in below codes per line\n",
    "\n",
    "gtex_pairwise_pearson_corr = pd.DataFrame(np.corrcoef(df_int,rowvar=False),\n",
    "                                          columns=df_int.columns,index=df_int.columns)\n",
    "#tcga_brca_df_normal_all is tcga normal dataframe\n",
    "\n",
    "tcga_normal_all_pearson_corr = pd.DataFrame(np.corrcoef(tcga_brca_df_normal_all,rowvar=False),\n",
    "                                          columns=tcga_brca_df_normal_all.columns,index=tcga_brca_df_normal_all.columns)\n",
    "\n",
    "#tcga_brca_df_primtum_all is tcga tumor dataframe\n",
    "\n",
    "tcga_primtum_all_pearson_corr = pd.DataFrame(np.corrcoef(tcga_brca_df_primtum_all,rowvar=False),\n",
    "                                          columns=tcga_brca_df_primtum_all.columns,\n",
    "                                             index=tcga_brca_df_primtum_all.columns)\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# top_genes = 2000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_genes=2000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_gtex_rank\talso_high_TCGA_normal\talso_high_TCGA_primtum\thigh_corr_all_3\n",
      "ITGA10\t2000\t981\t256\t164\n",
      "ITGAD\t2000\t418\t498\t285\n",
      "ITGAM\t2000\t1153\t771\t649\n",
      "ITGA3\t2000\t910\t364\t191\n",
      "ITGBL1\t2000\t743\t519\t217\n",
      "ITGAE\t2000\t727\t577\t126\n",
      "ITGA2\t2000\t1248\t241\t138\n",
      "ITGB3\t2000\t1091\t702\t535\n",
      "ITGA7\t2000\t1369\t585\t522\n",
      "ITGB8\t2000\t1225\t389\t288\n",
      "ITGAX\t2000\t562\t511\t331\n",
      "ITGAV\t2000\t1150\t1021\t817\n",
      "ITGB6\t2000\t1258\t302\t223\n",
      "ITGB7\t2000\t693\t239\t188\n",
      "ITGA5\t2000\t1123\t901\t674\n",
      "ITGB4\t2000\t1147\t510\t388\n",
      "ITGB2\t2000\t790\t765\t518\n",
      "ITGA6\t2000\t1070\t790\t506\n",
      "ITGA2B\t2000\t312\t375\t83\n",
      "ITGB1\t2000\t1193\t664\t471\n",
      "ITGAL\t2000\t713\t432\t301\n",
      "ITGA9\t2000\t856\t655\t335\n",
      "ITGB5\t2000\t954\t520\t309\n",
      "ITGA8\t2000\t788\t741\t476\n",
      "ITGA4\t2000\t734\t397\t327\n",
      "ITGA1\t2000\t1033\t802\t589\n",
      "ITGA11\t2000\t309\t242\t108\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>gtex_highly_correlated_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_TCGA_primtum</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>2000</td>\n",
       "      <td>693</td>\n",
       "      <td>239</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>2000</td>\n",
       "      <td>1248</td>\n",
       "      <td>241</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>2000</td>\n",
       "      <td>309</td>\n",
       "      <td>242</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>2000</td>\n",
       "      <td>981</td>\n",
       "      <td>256</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>2000</td>\n",
       "      <td>1258</td>\n",
       "      <td>302</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>2000</td>\n",
       "      <td>910</td>\n",
       "      <td>364</td>\n",
       "      <td>191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>2000</td>\n",
       "      <td>312</td>\n",
       "      <td>375</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>2000</td>\n",
       "      <td>1225</td>\n",
       "      <td>389</td>\n",
       "      <td>288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>2000</td>\n",
       "      <td>734</td>\n",
       "      <td>397</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>2000</td>\n",
       "      <td>713</td>\n",
       "      <td>432</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>2000</td>\n",
       "      <td>418</td>\n",
       "      <td>498</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>2000</td>\n",
       "      <td>1147</td>\n",
       "      <td>510</td>\n",
       "      <td>388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>2000</td>\n",
       "      <td>562</td>\n",
       "      <td>511</td>\n",
       "      <td>331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>2000</td>\n",
       "      <td>743</td>\n",
       "      <td>519</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>2000</td>\n",
       "      <td>954</td>\n",
       "      <td>520</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>2000</td>\n",
       "      <td>727</td>\n",
       "      <td>577</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>2000</td>\n",
       "      <td>1369</td>\n",
       "      <td>585</td>\n",
       "      <td>522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>2000</td>\n",
       "      <td>856</td>\n",
       "      <td>655</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>2000</td>\n",
       "      <td>1193</td>\n",
       "      <td>664</td>\n",
       "      <td>471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>2000</td>\n",
       "      <td>1091</td>\n",
       "      <td>702</td>\n",
       "      <td>535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>2000</td>\n",
       "      <td>788</td>\n",
       "      <td>741</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>2000</td>\n",
       "      <td>790</td>\n",
       "      <td>765</td>\n",
       "      <td>518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>2000</td>\n",
       "      <td>1153</td>\n",
       "      <td>771</td>\n",
       "      <td>649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>2000</td>\n",
       "      <td>1070</td>\n",
       "      <td>790</td>\n",
       "      <td>506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>2000</td>\n",
       "      <td>1033</td>\n",
       "      <td>802</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>2000</td>\n",
       "      <td>1123</td>\n",
       "      <td>901</td>\n",
       "      <td>674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>2000</td>\n",
       "      <td>1150</td>\n",
       "      <td>1021</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  gtex_highly_correlated_rank  also_high_TCGA_normal  \\\n",
       "13    ITGB7                         2000                    693   \n",
       "6     ITGA2                         2000                   1248   \n",
       "26   ITGA11                         2000                    309   \n",
       "0    ITGA10                         2000                    981   \n",
       "12    ITGB6                         2000                   1258   \n",
       "3     ITGA3                         2000                    910   \n",
       "18   ITGA2B                         2000                    312   \n",
       "9     ITGB8                         2000                   1225   \n",
       "24    ITGA4                         2000                    734   \n",
       "20    ITGAL                         2000                    713   \n",
       "1     ITGAD                         2000                    418   \n",
       "15    ITGB4                         2000                   1147   \n",
       "10    ITGAX                         2000                    562   \n",
       "4    ITGBL1                         2000                    743   \n",
       "22    ITGB5                         2000                    954   \n",
       "5     ITGAE                         2000                    727   \n",
       "8     ITGA7                         2000                   1369   \n",
       "21    ITGA9                         2000                    856   \n",
       "19    ITGB1                         2000                   1193   \n",
       "7     ITGB3                         2000                   1091   \n",
       "23    ITGA8                         2000                    788   \n",
       "16    ITGB2                         2000                    790   \n",
       "2     ITGAM                         2000                   1153   \n",
       "17    ITGA6                         2000                   1070   \n",
       "25    ITGA1                         2000                   1033   \n",
       "14    ITGA5                         2000                   1123   \n",
       "11    ITGAV                         2000                   1150   \n",
       "\n",
       "    also_high_TCGA_primtum  high_corr_all_3  \n",
       "13                     239              188  \n",
       "6                      241              138  \n",
       "26                     242              108  \n",
       "0                      256              164  \n",
       "12                     302              223  \n",
       "3                      364              191  \n",
       "18                     375               83  \n",
       "9                      389              288  \n",
       "24                     397              327  \n",
       "20                     432              301  \n",
       "1                      498              285  \n",
       "15                     510              388  \n",
       "10                     511              331  \n",
       "4                      519              217  \n",
       "22                     520              309  \n",
       "5                      577              126  \n",
       "8                      585              522  \n",
       "21                     655              335  \n",
       "19                     664              471  \n",
       "7                      702              535  \n",
       "23                     741              476  \n",
       "16                     765              518  \n",
       "2                      771              649  \n",
       "17                     790              506  \n",
       "25                     802              589  \n",
       "14                     901              674  \n",
       "11                    1021              817  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in GTEx - tracking those highly correlated genes across datasets\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "print('integrin','highly_correlated_gtex_rank','also_high_TCGA_normal','also_high_TCGA_primtum',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "gtex_highly_correlated=[]\n",
    "tcga_normal_highly_correlated=[]\n",
    "tcga_primtum_highly_correlated=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_gtex=correlation_selected_gene(df_int,itg,'pearson')\n",
    "    highly_corr_gtex=highly_corr_gtex.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_primtum = correlation_selected_gene(tcga_brca_df_primtum_all,\n",
    "                                                  itg,'pearson')\n",
    "    also_tcga_primtum = corr_tcga_primtum.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    high_corr_all_3 = highly_corr_gtex.index.intersection(also_tcga_normal.index).intersection(also_tcga_primtum.index)\n",
    "    \n",
    "    #changing the tcga normal and tcga primary tumor variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_gtex.index.intersection(also_tcga_normal.index)]\n",
    "    also_tcga_primtum = also_tcga_primtum[highly_corr_gtex.index.intersection(also_tcga_primtum.index)]\n",
    "    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    gtex_highly_correlated.append(highly_corr_gtex.shape[0])\n",
    "    tcga_normal_highly_correlated.append(also_tcga_normal.shape[0])\n",
    "    tcga_primtum_highly_correlated.append(also_tcga_primtum.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_gtex.shape[0], also_tcga_normal.shape[0], also_tcga_primtum.shape[0],\n",
    "          high_corr_all_3.shape[0],\n",
    "          sep='\\t')\n",
    "    \n",
    "gtex_corr_lost_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             gtex_highly_correlated,\n",
    "                                             tcga_normal_highly_correlated,\n",
    "                           tcga_primtum_highly_correlated,\n",
    "                                           highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'gtex_highly_correlated_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_TCGA_primtum',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_TCGA_primtum')\n",
    "del gtex_highly_correlated, tcga_normal_highly_correlated, tcga_primtum_highly_correlated, highly_correlated_all_3\n",
    "gtex_corr_lost_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_corr_lost_df.to_excel(f'rankbased_GTEx_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_TCGA_primtum_rank\talso_high_TCGA_normal\talso_high_GTEx\thigh_corr_all_3\n",
      "ITGA10\t2000\t367\t256\t164\n",
      "ITGAD\t2000\t972\t498\t285\n",
      "ITGAM\t2000\t980\t771\t649\n",
      "ITGA3\t2000\t376\t364\t191\n",
      "ITGBL1\t2000\t386\t519\t217\n",
      "ITGAE\t2000\t163\t577\t126\n",
      "ITGA2\t2000\t245\t241\t138\n",
      "ITGB3\t2000\t832\t702\t535\n",
      "ITGA7\t2000\t728\t585\t522\n",
      "ITGB8\t2000\t435\t389\t288\n",
      "ITGAX\t2000\t882\t511\t331\n",
      "ITGAV\t2000\t1291\t1021\t817\n",
      "ITGB6\t2000\t305\t302\t223\n",
      "ITGB7\t2000\t688\t239\t188\n",
      "ITGA5\t2000\t1037\t901\t674\n",
      "ITGB4\t2000\t564\t510\t388\n",
      "ITGB2\t2000\t861\t765\t518\n",
      "ITGA6\t2000\t793\t790\t506\n",
      "ITGA2B\t2000\t263\t375\t83\n",
      "ITGB1\t2000\t713\t664\t471\n",
      "ITGAL\t2000\t652\t432\t301\n",
      "ITGA9\t2000\t662\t655\t335\n",
      "ITGB5\t2000\t509\t520\t309\n",
      "ITGA8\t2000\t960\t741\t476\n",
      "ITGA4\t2000\t878\t397\t327\n",
      "ITGA1\t2000\t896\t802\t589\n",
      "ITGA11\t2000\t624\t242\t108\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>BRCA_high_corr_genes_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_GTEx</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>2000</td>\n",
       "      <td>688</td>\n",
       "      <td>239</td>\n",
       "      <td>188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>2000</td>\n",
       "      <td>245</td>\n",
       "      <td>241</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>2000</td>\n",
       "      <td>624</td>\n",
       "      <td>242</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>2000</td>\n",
       "      <td>367</td>\n",
       "      <td>256</td>\n",
       "      <td>164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>2000</td>\n",
       "      <td>305</td>\n",
       "      <td>302</td>\n",
       "      <td>223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>2000</td>\n",
       "      <td>376</td>\n",
       "      <td>364</td>\n",
       "      <td>191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>2000</td>\n",
       "      <td>263</td>\n",
       "      <td>375</td>\n",
       "      <td>83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>2000</td>\n",
       "      <td>435</td>\n",
       "      <td>389</td>\n",
       "      <td>288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>2000</td>\n",
       "      <td>878</td>\n",
       "      <td>397</td>\n",
       "      <td>327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>2000</td>\n",
       "      <td>652</td>\n",
       "      <td>432</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>2000</td>\n",
       "      <td>972</td>\n",
       "      <td>498</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>2000</td>\n",
       "      <td>564</td>\n",
       "      <td>510</td>\n",
       "      <td>388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>2000</td>\n",
       "      <td>882</td>\n",
       "      <td>511</td>\n",
       "      <td>331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>2000</td>\n",
       "      <td>386</td>\n",
       "      <td>519</td>\n",
       "      <td>217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>2000</td>\n",
       "      <td>509</td>\n",
       "      <td>520</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>2000</td>\n",
       "      <td>163</td>\n",
       "      <td>577</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>2000</td>\n",
       "      <td>728</td>\n",
       "      <td>585</td>\n",
       "      <td>522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>2000</td>\n",
       "      <td>662</td>\n",
       "      <td>655</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>2000</td>\n",
       "      <td>713</td>\n",
       "      <td>664</td>\n",
       "      <td>471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>2000</td>\n",
       "      <td>832</td>\n",
       "      <td>702</td>\n",
       "      <td>535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>2000</td>\n",
       "      <td>960</td>\n",
       "      <td>741</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>2000</td>\n",
       "      <td>861</td>\n",
       "      <td>765</td>\n",
       "      <td>518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>2000</td>\n",
       "      <td>980</td>\n",
       "      <td>771</td>\n",
       "      <td>649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>2000</td>\n",
       "      <td>793</td>\n",
       "      <td>790</td>\n",
       "      <td>506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>2000</td>\n",
       "      <td>896</td>\n",
       "      <td>802</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>2000</td>\n",
       "      <td>1037</td>\n",
       "      <td>901</td>\n",
       "      <td>674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>2000</td>\n",
       "      <td>1291</td>\n",
       "      <td>1021</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  BRCA_high_corr_genes_rank  also_high_TCGA_normal  also_high_GTEx  \\\n",
       "13    ITGB7                       2000                    688             239   \n",
       "6     ITGA2                       2000                    245             241   \n",
       "26   ITGA11                       2000                    624             242   \n",
       "0    ITGA10                       2000                    367             256   \n",
       "12    ITGB6                       2000                    305             302   \n",
       "3     ITGA3                       2000                    376             364   \n",
       "18   ITGA2B                       2000                    263             375   \n",
       "9     ITGB8                       2000                    435             389   \n",
       "24    ITGA4                       2000                    878             397   \n",
       "20    ITGAL                       2000                    652             432   \n",
       "1     ITGAD                       2000                    972             498   \n",
       "15    ITGB4                       2000                    564             510   \n",
       "10    ITGAX                       2000                    882             511   \n",
       "4    ITGBL1                       2000                    386             519   \n",
       "22    ITGB5                       2000                    509             520   \n",
       "5     ITGAE                       2000                    163             577   \n",
       "8     ITGA7                       2000                    728             585   \n",
       "21    ITGA9                       2000                    662             655   \n",
       "19    ITGB1                       2000                    713             664   \n",
       "7     ITGB3                       2000                    832             702   \n",
       "23    ITGA8                       2000                    960             741   \n",
       "16    ITGB2                       2000                    861             765   \n",
       "2     ITGAM                       2000                    980             771   \n",
       "17    ITGA6                       2000                    793             790   \n",
       "25    ITGA1                       2000                    896             802   \n",
       "14    ITGA5                       2000                   1037             901   \n",
       "11    ITGAV                       2000                   1291            1021   \n",
       "\n",
       "    high_corr_all_3  \n",
       "13              188  \n",
       "6               138  \n",
       "26              108  \n",
       "0               164  \n",
       "12              223  \n",
       "3               191  \n",
       "18               83  \n",
       "9               288  \n",
       "24              327  \n",
       "20              301  \n",
       "1               285  \n",
       "15              388  \n",
       "10              331  \n",
       "4               217  \n",
       "22              309  \n",
       "5               126  \n",
       "8               522  \n",
       "21              335  \n",
       "19              471  \n",
       "7               535  \n",
       "23              476  \n",
       "16              518  \n",
       "2               649  \n",
       "17              506  \n",
       "25              589  \n",
       "14              674  \n",
       "11              817  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in TCGA BRCA primary tumor,\n",
    "#tracking those highly correlated genes - what were those correlations in GTEx and TCGA normal\n",
    "\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "\n",
    "print('integrin','highly_correlated_TCGA_primtum_rank','also_high_TCGA_normal','also_high_GTEx',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "BRCA_high_corr_genes=[]\n",
    "also_BRCA_normal=[]\n",
    "also_breast_GTEx=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_BRCA=correlation_selected_gene(tcga_brca_df_primtum_all,itg,'pearson')\n",
    "    highly_corr_BRCA=highly_corr_BRCA.sort_values(ascending=False)[:top_genes]\n",
    "        \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    corr_gtex_normal = correlation_selected_gene(df_int,\n",
    "                                                  itg,'pearson')\n",
    "    also_gtex_normal = corr_gtex_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    high_corr_all_3 = highly_corr_BRCA.index.intersection(also_tcga_normal.index).intersection(also_gtex_normal.index)\n",
    "    \n",
    "    #changing the tcga normal and gtex variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_BRCA.index.intersection(also_tcga_normal.index)]\n",
    "    also_gtex_normal = also_gtex_normal[highly_corr_BRCA.index.intersection(also_gtex_normal.index)]    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    BRCA_high_corr_genes.append(highly_corr_BRCA.shape[0])\n",
    "    also_BRCA_normal.append(also_tcga_normal.shape[0])\n",
    "    also_breast_GTEx.append(also_gtex_normal.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_BRCA.shape[0], also_tcga_normal.shape[0], also_gtex_normal.shape[0],\n",
    "          high_corr_all_3.shape[0],sep='\\t')\n",
    "    \n",
    "tcga_primtum_gained_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             BRCA_high_corr_genes,\n",
    "                                             also_BRCA_normal,\n",
    "                           also_breast_GTEx,\n",
    "                                                 highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'BRCA_high_corr_genes_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_GTEx',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_GTEx')\n",
    "del integrins, BRCA_high_corr_genes, also_BRCA_normal, also_breast_GTEx, highly_correlated_all_3\n",
    "tcga_primtum_gained_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "tcga_primtum_gained_df.to_excel(f'rankbased_BRCA_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "del gtex_corr_lost_df, tcga_primtum_gained_df, top_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# top_genes = 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_genes=1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_gtex_rank\talso_high_TCGA_normal\talso_high_TCGA_primtum\thigh_corr_all_3\n",
      "ITGA10\t1000\t334\t51\t30\n",
      "ITGAD\t1000\t156\t214\t106\n",
      "ITGAM\t1000\t569\t374\t324\n",
      "ITGA3\t1000\t330\t108\t49\n",
      "ITGBL1\t1000\t295\t184\t67\n",
      "ITGAE\t1000\t231\t205\t29\n",
      "ITGA2\t1000\t538\t38\t16\n",
      "ITGB3\t1000\t446\t270\t200\n",
      "ITGA7\t1000\t659\t276\t233\n",
      "ITGB8\t1000\t586\t156\t106\n",
      "ITGAX\t1000\t206\t220\t139\n",
      "ITGAV\t1000\t498\t415\t301\n",
      "ITGB6\t1000\t619\t109\t74\n",
      "ITGB7\t1000\t229\t98\t84\n",
      "ITGA5\t1000\t469\t348\t231\n",
      "ITGB4\t1000\t537\t236\t156\n",
      "ITGB2\t1000\t439\t421\t297\n",
      "ITGA6\t1000\t382\t279\t140\n",
      "ITGA2B\t1000\t86\t117\t13\n",
      "ITGB1\t1000\t510\t256\t171\n",
      "ITGAL\t1000\t271\t183\t144\n",
      "ITGA9\t1000\t314\t205\t89\n",
      "ITGB5\t1000\t366\t209\t99\n",
      "ITGA8\t1000\t343\t288\t178\n",
      "ITGA4\t1000\t247\t97\t72\n",
      "ITGA1\t1000\t419\t332\t203\n",
      "ITGA11\t1000\t116\t93\t34\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>gtex_highly_correlated_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_TCGA_primtum</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>1000</td>\n",
       "      <td>538</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>1000</td>\n",
       "      <td>334</td>\n",
       "      <td>51</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>1000</td>\n",
       "      <td>116</td>\n",
       "      <td>93</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>1000</td>\n",
       "      <td>247</td>\n",
       "      <td>97</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>1000</td>\n",
       "      <td>229</td>\n",
       "      <td>98</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>1000</td>\n",
       "      <td>330</td>\n",
       "      <td>108</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>1000</td>\n",
       "      <td>619</td>\n",
       "      <td>109</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>1000</td>\n",
       "      <td>86</td>\n",
       "      <td>117</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>1000</td>\n",
       "      <td>586</td>\n",
       "      <td>156</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>1000</td>\n",
       "      <td>271</td>\n",
       "      <td>183</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>1000</td>\n",
       "      <td>295</td>\n",
       "      <td>184</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>1000</td>\n",
       "      <td>231</td>\n",
       "      <td>205</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>1000</td>\n",
       "      <td>314</td>\n",
       "      <td>205</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>1000</td>\n",
       "      <td>366</td>\n",
       "      <td>209</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>1000</td>\n",
       "      <td>156</td>\n",
       "      <td>214</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>1000</td>\n",
       "      <td>206</td>\n",
       "      <td>220</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>1000</td>\n",
       "      <td>537</td>\n",
       "      <td>236</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>1000</td>\n",
       "      <td>510</td>\n",
       "      <td>256</td>\n",
       "      <td>171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>1000</td>\n",
       "      <td>446</td>\n",
       "      <td>270</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>1000</td>\n",
       "      <td>659</td>\n",
       "      <td>276</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>1000</td>\n",
       "      <td>382</td>\n",
       "      <td>279</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>1000</td>\n",
       "      <td>343</td>\n",
       "      <td>288</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>1000</td>\n",
       "      <td>419</td>\n",
       "      <td>332</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>1000</td>\n",
       "      <td>469</td>\n",
       "      <td>348</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>1000</td>\n",
       "      <td>569</td>\n",
       "      <td>374</td>\n",
       "      <td>324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>1000</td>\n",
       "      <td>498</td>\n",
       "      <td>415</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>1000</td>\n",
       "      <td>439</td>\n",
       "      <td>421</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  gtex_highly_correlated_rank  also_high_TCGA_normal  \\\n",
       "6     ITGA2                         1000                    538   \n",
       "0    ITGA10                         1000                    334   \n",
       "26   ITGA11                         1000                    116   \n",
       "24    ITGA4                         1000                    247   \n",
       "13    ITGB7                         1000                    229   \n",
       "3     ITGA3                         1000                    330   \n",
       "12    ITGB6                         1000                    619   \n",
       "18   ITGA2B                         1000                     86   \n",
       "9     ITGB8                         1000                    586   \n",
       "20    ITGAL                         1000                    271   \n",
       "4    ITGBL1                         1000                    295   \n",
       "5     ITGAE                         1000                    231   \n",
       "21    ITGA9                         1000                    314   \n",
       "22    ITGB5                         1000                    366   \n",
       "1     ITGAD                         1000                    156   \n",
       "10    ITGAX                         1000                    206   \n",
       "15    ITGB4                         1000                    537   \n",
       "19    ITGB1                         1000                    510   \n",
       "7     ITGB3                         1000                    446   \n",
       "8     ITGA7                         1000                    659   \n",
       "17    ITGA6                         1000                    382   \n",
       "23    ITGA8                         1000                    343   \n",
       "25    ITGA1                         1000                    419   \n",
       "14    ITGA5                         1000                    469   \n",
       "2     ITGAM                         1000                    569   \n",
       "11    ITGAV                         1000                    498   \n",
       "16    ITGB2                         1000                    439   \n",
       "\n",
       "    also_high_TCGA_primtum  high_corr_all_3  \n",
       "6                       38               16  \n",
       "0                       51               30  \n",
       "26                      93               34  \n",
       "24                      97               72  \n",
       "13                      98               84  \n",
       "3                      108               49  \n",
       "12                     109               74  \n",
       "18                     117               13  \n",
       "9                      156              106  \n",
       "20                     183              144  \n",
       "4                      184               67  \n",
       "5                      205               29  \n",
       "21                     205               89  \n",
       "22                     209               99  \n",
       "1                      214              106  \n",
       "10                     220              139  \n",
       "15                     236              156  \n",
       "19                     256              171  \n",
       "7                      270              200  \n",
       "8                      276              233  \n",
       "17                     279              140  \n",
       "23                     288              178  \n",
       "25                     332              203  \n",
       "14                     348              231  \n",
       "2                      374              324  \n",
       "11                     415              301  \n",
       "16                     421              297  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in GTEx - tracking those highly correlated genes across datasets\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "print('integrin','highly_correlated_gtex_rank','also_high_TCGA_normal','also_high_TCGA_primtum',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "gtex_highly_correlated=[]\n",
    "tcga_normal_highly_correlated=[]\n",
    "tcga_primtum_highly_correlated=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_gtex=correlation_selected_gene(df_int,itg,'pearson')\n",
    "    highly_corr_gtex=highly_corr_gtex.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_primtum = correlation_selected_gene(tcga_brca_df_primtum_all,\n",
    "                                                  itg,'pearson')\n",
    "    also_tcga_primtum = corr_tcga_primtum.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    high_corr_all_3 = highly_corr_gtex.index.intersection(also_tcga_normal.index).intersection(also_tcga_primtum.index)\n",
    "    \n",
    "    #changing the tcga normal and tcga primary tumor variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_gtex.index.intersection(also_tcga_normal.index)]\n",
    "    also_tcga_primtum = also_tcga_primtum[highly_corr_gtex.index.intersection(also_tcga_primtum.index)]\n",
    "    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    gtex_highly_correlated.append(highly_corr_gtex.shape[0])\n",
    "    tcga_normal_highly_correlated.append(also_tcga_normal.shape[0])\n",
    "    tcga_primtum_highly_correlated.append(also_tcga_primtum.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_gtex.shape[0], also_tcga_normal.shape[0], also_tcga_primtum.shape[0],\n",
    "          high_corr_all_3.shape[0],\n",
    "          sep='\\t')\n",
    "    \n",
    "gtex_corr_lost_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             gtex_highly_correlated,\n",
    "                                             tcga_normal_highly_correlated,\n",
    "                           tcga_primtum_highly_correlated,\n",
    "                                           highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'gtex_highly_correlated_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_TCGA_primtum',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_TCGA_primtum')\n",
    "del gtex_highly_correlated, tcga_normal_highly_correlated, tcga_primtum_highly_correlated, highly_correlated_all_3\n",
    "gtex_corr_lost_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_corr_lost_df.to_excel(f'rankbased_GTEx_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_TCGA_primtum_rank\talso_high_TCGA_normal\talso_high_GTEx\thigh_corr_all_3\n",
      "ITGA10\t1000\t130\t51\t30\n",
      "ITGAD\t1000\t449\t214\t106\n",
      "ITGAM\t1000\t479\t374\t324\n",
      "ITGA3\t1000\t143\t108\t49\n",
      "ITGBL1\t1000\t119\t184\t67\n",
      "ITGAE\t1000\t39\t205\t29\n",
      "ITGA2\t1000\t38\t38\t16\n",
      "ITGB3\t1000\t350\t270\t200\n",
      "ITGA7\t1000\t332\t276\t233\n",
      "ITGB8\t1000\t160\t156\t106\n",
      "ITGAX\t1000\t467\t220\t139\n",
      "ITGAV\t1000\t519\t415\t301\n",
      "ITGB6\t1000\t106\t109\t74\n",
      "ITGB7\t1000\t364\t98\t84\n",
      "ITGA5\t1000\t414\t348\t231\n",
      "ITGB4\t1000\t236\t236\t156\n",
      "ITGB2\t1000\t440\t421\t297\n",
      "ITGA6\t1000\t269\t279\t140\n",
      "ITGA2B\t1000\t67\t117\t13\n",
      "ITGB1\t1000\t264\t256\t171\n",
      "ITGAL\t1000\t384\t183\t144\n",
      "ITGA9\t1000\t218\t205\t89\n",
      "ITGB5\t1000\t198\t209\t99\n",
      "ITGA8\t1000\t444\t288\t178\n",
      "ITGA4\t1000\t277\t97\t72\n",
      "ITGA1\t1000\t348\t332\t203\n",
      "ITGA11\t1000\t220\t93\t34\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>BRCA_high_corr_genes_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_GTEx</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>1000</td>\n",
       "      <td>38</td>\n",
       "      <td>38</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>1000</td>\n",
       "      <td>130</td>\n",
       "      <td>51</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>1000</td>\n",
       "      <td>220</td>\n",
       "      <td>93</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>1000</td>\n",
       "      <td>277</td>\n",
       "      <td>97</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>1000</td>\n",
       "      <td>364</td>\n",
       "      <td>98</td>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>1000</td>\n",
       "      <td>143</td>\n",
       "      <td>108</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>1000</td>\n",
       "      <td>106</td>\n",
       "      <td>109</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>1000</td>\n",
       "      <td>67</td>\n",
       "      <td>117</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>1000</td>\n",
       "      <td>160</td>\n",
       "      <td>156</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>1000</td>\n",
       "      <td>384</td>\n",
       "      <td>183</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>1000</td>\n",
       "      <td>119</td>\n",
       "      <td>184</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>1000</td>\n",
       "      <td>39</td>\n",
       "      <td>205</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>1000</td>\n",
       "      <td>218</td>\n",
       "      <td>205</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>1000</td>\n",
       "      <td>198</td>\n",
       "      <td>209</td>\n",
       "      <td>99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>1000</td>\n",
       "      <td>449</td>\n",
       "      <td>214</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>1000</td>\n",
       "      <td>467</td>\n",
       "      <td>220</td>\n",
       "      <td>139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>1000</td>\n",
       "      <td>236</td>\n",
       "      <td>236</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>1000</td>\n",
       "      <td>264</td>\n",
       "      <td>256</td>\n",
       "      <td>171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>1000</td>\n",
       "      <td>350</td>\n",
       "      <td>270</td>\n",
       "      <td>200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>1000</td>\n",
       "      <td>332</td>\n",
       "      <td>276</td>\n",
       "      <td>233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>1000</td>\n",
       "      <td>269</td>\n",
       "      <td>279</td>\n",
       "      <td>140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>1000</td>\n",
       "      <td>444</td>\n",
       "      <td>288</td>\n",
       "      <td>178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>1000</td>\n",
       "      <td>348</td>\n",
       "      <td>332</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>1000</td>\n",
       "      <td>414</td>\n",
       "      <td>348</td>\n",
       "      <td>231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>1000</td>\n",
       "      <td>479</td>\n",
       "      <td>374</td>\n",
       "      <td>324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>1000</td>\n",
       "      <td>519</td>\n",
       "      <td>415</td>\n",
       "      <td>301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>1000</td>\n",
       "      <td>440</td>\n",
       "      <td>421</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  BRCA_high_corr_genes_rank  also_high_TCGA_normal  also_high_GTEx  \\\n",
       "6     ITGA2                       1000                     38              38   \n",
       "0    ITGA10                       1000                    130              51   \n",
       "26   ITGA11                       1000                    220              93   \n",
       "24    ITGA4                       1000                    277              97   \n",
       "13    ITGB7                       1000                    364              98   \n",
       "3     ITGA3                       1000                    143             108   \n",
       "12    ITGB6                       1000                    106             109   \n",
       "18   ITGA2B                       1000                     67             117   \n",
       "9     ITGB8                       1000                    160             156   \n",
       "20    ITGAL                       1000                    384             183   \n",
       "4    ITGBL1                       1000                    119             184   \n",
       "5     ITGAE                       1000                     39             205   \n",
       "21    ITGA9                       1000                    218             205   \n",
       "22    ITGB5                       1000                    198             209   \n",
       "1     ITGAD                       1000                    449             214   \n",
       "10    ITGAX                       1000                    467             220   \n",
       "15    ITGB4                       1000                    236             236   \n",
       "19    ITGB1                       1000                    264             256   \n",
       "7     ITGB3                       1000                    350             270   \n",
       "8     ITGA7                       1000                    332             276   \n",
       "17    ITGA6                       1000                    269             279   \n",
       "23    ITGA8                       1000                    444             288   \n",
       "25    ITGA1                       1000                    348             332   \n",
       "14    ITGA5                       1000                    414             348   \n",
       "2     ITGAM                       1000                    479             374   \n",
       "11    ITGAV                       1000                    519             415   \n",
       "16    ITGB2                       1000                    440             421   \n",
       "\n",
       "    high_corr_all_3  \n",
       "6                16  \n",
       "0                30  \n",
       "26               34  \n",
       "24               72  \n",
       "13               84  \n",
       "3                49  \n",
       "12               74  \n",
       "18               13  \n",
       "9               106  \n",
       "20              144  \n",
       "4                67  \n",
       "5                29  \n",
       "21               89  \n",
       "22               99  \n",
       "1               106  \n",
       "10              139  \n",
       "15              156  \n",
       "19              171  \n",
       "7               200  \n",
       "8               233  \n",
       "17              140  \n",
       "23              178  \n",
       "25              203  \n",
       "14              231  \n",
       "2               324  \n",
       "11              301  \n",
       "16              297  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in TCGA BRCA primary tumor,\n",
    "#tracking those highly correlated genes - what were those correlations in GTEx and TCGA normal\n",
    "\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "\n",
    "print('integrin','highly_correlated_TCGA_primtum_rank','also_high_TCGA_normal','also_high_GTEx',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "BRCA_high_corr_genes=[]\n",
    "also_BRCA_normal=[]\n",
    "also_breast_GTEx=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_BRCA=correlation_selected_gene(tcga_brca_df_primtum_all,itg,'pearson')\n",
    "    highly_corr_BRCA=highly_corr_BRCA.sort_values(ascending=False)[:top_genes]\n",
    "        \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    corr_gtex_normal = correlation_selected_gene(df_int,\n",
    "                                                  itg,'pearson')\n",
    "    also_gtex_normal = corr_gtex_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    high_corr_all_3 = highly_corr_BRCA.index.intersection(also_tcga_normal.index).intersection(also_gtex_normal.index)\n",
    "    \n",
    "    #changing the tcga normal and gtex variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_BRCA.index.intersection(also_tcga_normal.index)]\n",
    "    also_gtex_normal = also_gtex_normal[highly_corr_BRCA.index.intersection(also_gtex_normal.index)]    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    BRCA_high_corr_genes.append(highly_corr_BRCA.shape[0])\n",
    "    also_BRCA_normal.append(also_tcga_normal.shape[0])\n",
    "    also_breast_GTEx.append(also_gtex_normal.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_BRCA.shape[0], also_tcga_normal.shape[0], also_gtex_normal.shape[0],\n",
    "          high_corr_all_3.shape[0],sep='\\t')\n",
    "    \n",
    "tcga_primtum_gained_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             BRCA_high_corr_genes,\n",
    "                                             also_BRCA_normal,\n",
    "                           also_breast_GTEx,\n",
    "                                                 highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'BRCA_high_corr_genes_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_GTEx',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_GTEx')\n",
    "del integrins, BRCA_high_corr_genes, also_BRCA_normal, also_breast_GTEx, highly_correlated_all_3\n",
    "tcga_primtum_gained_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "tcga_primtum_gained_df.to_excel(f'rankbased_BRCA_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "del gtex_corr_lost_df, tcga_primtum_gained_df, top_genes"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# top_genes = 4000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_genes=4000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_gtex_rank\talso_high_TCGA_normal\talso_high_TCGA_primtum\thigh_corr_all_3\n",
      "ITGA10\t4000\t2553\t1074\t770\n",
      "ITGAD\t4000\t1149\t1242\t749\n",
      "ITGAM\t4000\t2542\t1736\t1463\n",
      "ITGA3\t4000\t2275\t1211\t714\n",
      "ITGBL1\t4000\t2022\t1580\t906\n",
      "ITGAE\t4000\t1978\t1498\t566\n",
      "ITGA2\t4000\t2707\t1218\t860\n",
      "ITGB3\t4000\t2574\t1835\t1515\n",
      "ITGA7\t4000\t2817\t1280\t1160\n",
      "ITGB8\t4000\t2653\t1191\t947\n",
      "ITGAX\t4000\t1562\t1415\t846\n",
      "ITGAV\t4000\t2691\t2499\t2169\n",
      "ITGB6\t4000\t2656\t980\t707\n",
      "ITGB7\t4000\t2014\t788\t614\n",
      "ITGA5\t4000\t2513\t2129\t1692\n",
      "ITGB4\t4000\t2567\t1184\t943\n",
      "ITGB2\t4000\t1711\t1651\t1116\n",
      "ITGA6\t4000\t2581\t2045\t1605\n",
      "ITGA2B\t4000\t1100\t1181\t402\n",
      "ITGB1\t4000\t2737\t1864\t1483\n",
      "ITGAL\t4000\t2026\t1323\t785\n",
      "ITGA9\t4000\t2239\t1937\t1308\n",
      "ITGB5\t4000\t2281\t1393\t938\n",
      "ITGA8\t4000\t1967\t1873\t1304\n",
      "ITGA4\t4000\t2032\t1287\t1168\n",
      "ITGA1\t4000\t2504\t1990\t1593\n",
      "ITGA11\t4000\t1111\t933\t489\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>gtex_highly_correlated_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_TCGA_primtum</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>4000</td>\n",
       "      <td>2014</td>\n",
       "      <td>788</td>\n",
       "      <td>614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>4000</td>\n",
       "      <td>1111</td>\n",
       "      <td>933</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>4000</td>\n",
       "      <td>2656</td>\n",
       "      <td>980</td>\n",
       "      <td>707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>4000</td>\n",
       "      <td>2553</td>\n",
       "      <td>1074</td>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>4000</td>\n",
       "      <td>1100</td>\n",
       "      <td>1181</td>\n",
       "      <td>402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>4000</td>\n",
       "      <td>2567</td>\n",
       "      <td>1184</td>\n",
       "      <td>943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>4000</td>\n",
       "      <td>2653</td>\n",
       "      <td>1191</td>\n",
       "      <td>947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>4000</td>\n",
       "      <td>2275</td>\n",
       "      <td>1211</td>\n",
       "      <td>714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>4000</td>\n",
       "      <td>2707</td>\n",
       "      <td>1218</td>\n",
       "      <td>860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>4000</td>\n",
       "      <td>1149</td>\n",
       "      <td>1242</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>4000</td>\n",
       "      <td>2817</td>\n",
       "      <td>1280</td>\n",
       "      <td>1160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>4000</td>\n",
       "      <td>2032</td>\n",
       "      <td>1287</td>\n",
       "      <td>1168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>4000</td>\n",
       "      <td>2026</td>\n",
       "      <td>1323</td>\n",
       "      <td>785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>4000</td>\n",
       "      <td>2281</td>\n",
       "      <td>1393</td>\n",
       "      <td>938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>4000</td>\n",
       "      <td>1562</td>\n",
       "      <td>1415</td>\n",
       "      <td>846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>4000</td>\n",
       "      <td>1978</td>\n",
       "      <td>1498</td>\n",
       "      <td>566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>4000</td>\n",
       "      <td>2022</td>\n",
       "      <td>1580</td>\n",
       "      <td>906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>4000</td>\n",
       "      <td>1711</td>\n",
       "      <td>1651</td>\n",
       "      <td>1116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>4000</td>\n",
       "      <td>2542</td>\n",
       "      <td>1736</td>\n",
       "      <td>1463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>4000</td>\n",
       "      <td>2574</td>\n",
       "      <td>1835</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>4000</td>\n",
       "      <td>2737</td>\n",
       "      <td>1864</td>\n",
       "      <td>1483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>4000</td>\n",
       "      <td>1967</td>\n",
       "      <td>1873</td>\n",
       "      <td>1304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>4000</td>\n",
       "      <td>2239</td>\n",
       "      <td>1937</td>\n",
       "      <td>1308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>4000</td>\n",
       "      <td>2504</td>\n",
       "      <td>1990</td>\n",
       "      <td>1593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>4000</td>\n",
       "      <td>2581</td>\n",
       "      <td>2045</td>\n",
       "      <td>1605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>4000</td>\n",
       "      <td>2513</td>\n",
       "      <td>2129</td>\n",
       "      <td>1692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>4000</td>\n",
       "      <td>2691</td>\n",
       "      <td>2499</td>\n",
       "      <td>2169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  gtex_highly_correlated_rank  also_high_TCGA_normal  \\\n",
       "13    ITGB7                         4000                   2014   \n",
       "26   ITGA11                         4000                   1111   \n",
       "12    ITGB6                         4000                   2656   \n",
       "0    ITGA10                         4000                   2553   \n",
       "18   ITGA2B                         4000                   1100   \n",
       "15    ITGB4                         4000                   2567   \n",
       "9     ITGB8                         4000                   2653   \n",
       "3     ITGA3                         4000                   2275   \n",
       "6     ITGA2                         4000                   2707   \n",
       "1     ITGAD                         4000                   1149   \n",
       "8     ITGA7                         4000                   2817   \n",
       "24    ITGA4                         4000                   2032   \n",
       "20    ITGAL                         4000                   2026   \n",
       "22    ITGB5                         4000                   2281   \n",
       "10    ITGAX                         4000                   1562   \n",
       "5     ITGAE                         4000                   1978   \n",
       "4    ITGBL1                         4000                   2022   \n",
       "16    ITGB2                         4000                   1711   \n",
       "2     ITGAM                         4000                   2542   \n",
       "7     ITGB3                         4000                   2574   \n",
       "19    ITGB1                         4000                   2737   \n",
       "23    ITGA8                         4000                   1967   \n",
       "21    ITGA9                         4000                   2239   \n",
       "25    ITGA1                         4000                   2504   \n",
       "17    ITGA6                         4000                   2581   \n",
       "14    ITGA5                         4000                   2513   \n",
       "11    ITGAV                         4000                   2691   \n",
       "\n",
       "    also_high_TCGA_primtum  high_corr_all_3  \n",
       "13                     788              614  \n",
       "26                     933              489  \n",
       "12                     980              707  \n",
       "0                     1074              770  \n",
       "18                    1181              402  \n",
       "15                    1184              943  \n",
       "9                     1191              947  \n",
       "3                     1211              714  \n",
       "6                     1218              860  \n",
       "1                     1242              749  \n",
       "8                     1280             1160  \n",
       "24                    1287             1168  \n",
       "20                    1323              785  \n",
       "22                    1393              938  \n",
       "10                    1415              846  \n",
       "5                     1498              566  \n",
       "4                     1580              906  \n",
       "16                    1651             1116  \n",
       "2                     1736             1463  \n",
       "7                     1835             1515  \n",
       "19                    1864             1483  \n",
       "23                    1873             1304  \n",
       "21                    1937             1308  \n",
       "25                    1990             1593  \n",
       "17                    2045             1605  \n",
       "14                    2129             1692  \n",
       "11                    2499             2169  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in GTEx - tracking those highly correlated genes across datasets\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "print('integrin','highly_correlated_gtex_rank','also_high_TCGA_normal','also_high_TCGA_primtum',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "gtex_highly_correlated=[]\n",
    "tcga_normal_highly_correlated=[]\n",
    "tcga_primtum_highly_correlated=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_gtex=correlation_selected_gene(df_int,itg,'pearson')\n",
    "    highly_corr_gtex=highly_corr_gtex.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    corr_tcga_primtum = correlation_selected_gene(tcga_brca_df_primtum_all,\n",
    "                                                  itg,'pearson')\n",
    "    also_tcga_primtum = corr_tcga_primtum.sort_values(ascending=False)[:top_genes]\n",
    "    \n",
    "    high_corr_all_3 = highly_corr_gtex.index.intersection(also_tcga_normal.index).intersection(also_tcga_primtum.index)\n",
    "    \n",
    "    #changing the tcga normal and tcga primary tumor variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_gtex.index.intersection(also_tcga_normal.index)]\n",
    "    also_tcga_primtum = also_tcga_primtum[highly_corr_gtex.index.intersection(also_tcga_primtum.index)]\n",
    "    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    gtex_highly_correlated.append(highly_corr_gtex.shape[0])\n",
    "    tcga_normal_highly_correlated.append(also_tcga_normal.shape[0])\n",
    "    tcga_primtum_highly_correlated.append(also_tcga_primtum.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_gtex.shape[0], also_tcga_normal.shape[0], also_tcga_primtum.shape[0],\n",
    "          high_corr_all_3.shape[0],\n",
    "          sep='\\t')\n",
    "    \n",
    "gtex_corr_lost_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             gtex_highly_correlated,\n",
    "                                             tcga_normal_highly_correlated,\n",
    "                           tcga_primtum_highly_correlated,\n",
    "                                           highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'gtex_highly_correlated_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_TCGA_primtum',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_TCGA_primtum')\n",
    "del gtex_highly_correlated, tcga_normal_highly_correlated, tcga_primtum_highly_correlated, highly_correlated_all_3\n",
    "gtex_corr_lost_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_corr_lost_df.to_excel(f'rankbased_GTEx_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "integrin\thighly_correlated_TCGA_primtum_rank\talso_high_TCGA_normal\talso_high_GTEx\thigh_corr_all_3\n",
      "ITGA10\t4000\t1203\t1074\t770\n",
      "ITGAD\t4000\t2116\t1242\t749\n",
      "ITGAM\t4000\t2093\t1736\t1463\n",
      "ITGA3\t4000\t1120\t1211\t714\n",
      "ITGBL1\t4000\t1408\t1580\t906\n",
      "ITGAE\t4000\t758\t1498\t566\n",
      "ITGA2\t4000\t1295\t1218\t860\n",
      "ITGB3\t4000\t2148\t1835\t1515\n",
      "ITGA7\t4000\t1599\t1280\t1160\n",
      "ITGB8\t4000\t1418\t1191\t947\n",
      "ITGAX\t4000\t1720\t1415\t846\n",
      "ITGAV\t4000\t2996\t2499\t2169\n",
      "ITGB6\t4000\t1001\t980\t707\n",
      "ITGB7\t4000\t1510\t788\t614\n",
      "ITGA5\t4000\t2280\t2129\t1692\n",
      "ITGB4\t4000\t1368\t1184\t943\n",
      "ITGB2\t4000\t1936\t1651\t1116\n",
      "ITGA6\t4000\t2205\t2045\t1605\n",
      "ITGA2B\t4000\t975\t1181\t402\n",
      "ITGB1\t4000\t2063\t1864\t1483\n",
      "ITGAL\t4000\t1345\t1323\t785\n",
      "ITGA9\t4000\t2066\t1937\t1308\n",
      "ITGB5\t4000\t1443\t1393\t938\n",
      "ITGA8\t4000\t2177\t1873\t1304\n",
      "ITGA4\t4000\t2372\t1287\t1168\n",
      "ITGA1\t4000\t2315\t1990\t1593\n",
      "ITGA11\t4000\t1810\t933\t489\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integrin</th>\n",
       "      <th>BRCA_high_corr_genes_rank</th>\n",
       "      <th>also_high_TCGA_normal</th>\n",
       "      <th>also_high_GTEx</th>\n",
       "      <th>high_corr_all_3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ITGB7</td>\n",
       "      <td>4000</td>\n",
       "      <td>1510</td>\n",
       "      <td>788</td>\n",
       "      <td>614</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ITGA11</td>\n",
       "      <td>4000</td>\n",
       "      <td>1810</td>\n",
       "      <td>933</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ITGB6</td>\n",
       "      <td>4000</td>\n",
       "      <td>1001</td>\n",
       "      <td>980</td>\n",
       "      <td>707</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ITGA10</td>\n",
       "      <td>4000</td>\n",
       "      <td>1203</td>\n",
       "      <td>1074</td>\n",
       "      <td>770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ITGA2B</td>\n",
       "      <td>4000</td>\n",
       "      <td>975</td>\n",
       "      <td>1181</td>\n",
       "      <td>402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ITGB4</td>\n",
       "      <td>4000</td>\n",
       "      <td>1368</td>\n",
       "      <td>1184</td>\n",
       "      <td>943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ITGB8</td>\n",
       "      <td>4000</td>\n",
       "      <td>1418</td>\n",
       "      <td>1191</td>\n",
       "      <td>947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ITGA3</td>\n",
       "      <td>4000</td>\n",
       "      <td>1120</td>\n",
       "      <td>1211</td>\n",
       "      <td>714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ITGA2</td>\n",
       "      <td>4000</td>\n",
       "      <td>1295</td>\n",
       "      <td>1218</td>\n",
       "      <td>860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ITGAD</td>\n",
       "      <td>4000</td>\n",
       "      <td>2116</td>\n",
       "      <td>1242</td>\n",
       "      <td>749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ITGA7</td>\n",
       "      <td>4000</td>\n",
       "      <td>1599</td>\n",
       "      <td>1280</td>\n",
       "      <td>1160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ITGA4</td>\n",
       "      <td>4000</td>\n",
       "      <td>2372</td>\n",
       "      <td>1287</td>\n",
       "      <td>1168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ITGAL</td>\n",
       "      <td>4000</td>\n",
       "      <td>1345</td>\n",
       "      <td>1323</td>\n",
       "      <td>785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ITGB5</td>\n",
       "      <td>4000</td>\n",
       "      <td>1443</td>\n",
       "      <td>1393</td>\n",
       "      <td>938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ITGAX</td>\n",
       "      <td>4000</td>\n",
       "      <td>1720</td>\n",
       "      <td>1415</td>\n",
       "      <td>846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ITGAE</td>\n",
       "      <td>4000</td>\n",
       "      <td>758</td>\n",
       "      <td>1498</td>\n",
       "      <td>566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>4000</td>\n",
       "      <td>1408</td>\n",
       "      <td>1580</td>\n",
       "      <td>906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ITGB2</td>\n",
       "      <td>4000</td>\n",
       "      <td>1936</td>\n",
       "      <td>1651</td>\n",
       "      <td>1116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ITGAM</td>\n",
       "      <td>4000</td>\n",
       "      <td>2093</td>\n",
       "      <td>1736</td>\n",
       "      <td>1463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ITGB3</td>\n",
       "      <td>4000</td>\n",
       "      <td>2148</td>\n",
       "      <td>1835</td>\n",
       "      <td>1515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ITGB1</td>\n",
       "      <td>4000</td>\n",
       "      <td>2063</td>\n",
       "      <td>1864</td>\n",
       "      <td>1483</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ITGA8</td>\n",
       "      <td>4000</td>\n",
       "      <td>2177</td>\n",
       "      <td>1873</td>\n",
       "      <td>1304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ITGA9</td>\n",
       "      <td>4000</td>\n",
       "      <td>2066</td>\n",
       "      <td>1937</td>\n",
       "      <td>1308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ITGA1</td>\n",
       "      <td>4000</td>\n",
       "      <td>2315</td>\n",
       "      <td>1990</td>\n",
       "      <td>1593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ITGA6</td>\n",
       "      <td>4000</td>\n",
       "      <td>2205</td>\n",
       "      <td>2045</td>\n",
       "      <td>1605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ITGA5</td>\n",
       "      <td>4000</td>\n",
       "      <td>2280</td>\n",
       "      <td>2129</td>\n",
       "      <td>1692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ITGAV</td>\n",
       "      <td>4000</td>\n",
       "      <td>2996</td>\n",
       "      <td>2499</td>\n",
       "      <td>2169</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   integrin  BRCA_high_corr_genes_rank  also_high_TCGA_normal  also_high_GTEx  \\\n",
       "13    ITGB7                       4000                   1510             788   \n",
       "26   ITGA11                       4000                   1810             933   \n",
       "12    ITGB6                       4000                   1001             980   \n",
       "0    ITGA10                       4000                   1203            1074   \n",
       "18   ITGA2B                       4000                    975            1181   \n",
       "15    ITGB4                       4000                   1368            1184   \n",
       "9     ITGB8                       4000                   1418            1191   \n",
       "3     ITGA3                       4000                   1120            1211   \n",
       "6     ITGA2                       4000                   1295            1218   \n",
       "1     ITGAD                       4000                   2116            1242   \n",
       "8     ITGA7                       4000                   1599            1280   \n",
       "24    ITGA4                       4000                   2372            1287   \n",
       "20    ITGAL                       4000                   1345            1323   \n",
       "22    ITGB5                       4000                   1443            1393   \n",
       "10    ITGAX                       4000                   1720            1415   \n",
       "5     ITGAE                       4000                    758            1498   \n",
       "4    ITGBL1                       4000                   1408            1580   \n",
       "16    ITGB2                       4000                   1936            1651   \n",
       "2     ITGAM                       4000                   2093            1736   \n",
       "7     ITGB3                       4000                   2148            1835   \n",
       "19    ITGB1                       4000                   2063            1864   \n",
       "23    ITGA8                       4000                   2177            1873   \n",
       "21    ITGA9                       4000                   2066            1937   \n",
       "25    ITGA1                       4000                   2315            1990   \n",
       "17    ITGA6                       4000                   2205            2045   \n",
       "14    ITGA5                       4000                   2280            2129   \n",
       "11    ITGAV                       4000                   2996            2499   \n",
       "\n",
       "    high_corr_all_3  \n",
       "13              614  \n",
       "26              489  \n",
       "12              707  \n",
       "0               770  \n",
       "18              402  \n",
       "15              943  \n",
       "9               947  \n",
       "3               714  \n",
       "6               860  \n",
       "1               749  \n",
       "8              1160  \n",
       "24             1168  \n",
       "20              785  \n",
       "22              938  \n",
       "10              846  \n",
       "5               566  \n",
       "4               906  \n",
       "16             1116  \n",
       "2              1463  \n",
       "7              1515  \n",
       "19             1483  \n",
       "23             1304  \n",
       "21             1308  \n",
       "25             1593  \n",
       "17             1605  \n",
       "14             1692  \n",
       "11             2169  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#this is for genes highly correlated w integrins in TCGA BRCA primary tumor,\n",
    "#tracking those highly correlated genes - what were those correlations in GTEx and TCGA normal\n",
    "\n",
    "#BE careful with edit and replace - check entire cell when changing\n",
    "\n",
    "print('integrin','highly_correlated_TCGA_primtum_rank','also_high_TCGA_normal','also_high_GTEx',\n",
    "      'high_corr_all_3',sep='\\t')\n",
    "\n",
    "integrins=[]\n",
    "BRCA_high_corr_genes=[]\n",
    "also_BRCA_normal=[]\n",
    "also_breast_GTEx=[]\n",
    "highly_correlated_all_3=[]\n",
    "\n",
    "for itg in itg_list:\n",
    "    highly_corr_BRCA=correlation_selected_gene(tcga_brca_df_primtum_all,itg,'pearson')\n",
    "    highly_corr_BRCA=highly_corr_BRCA.sort_values(ascending=False)[:top_genes]\n",
    "        \n",
    "    corr_tcga_normal = correlation_selected_gene(tcga_brca_df_normal_all,\n",
    "                                                 itg,'pearson')\n",
    "    also_tcga_normal = corr_tcga_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    corr_gtex_normal = correlation_selected_gene(df_int,\n",
    "                                                  itg,'pearson')\n",
    "    also_gtex_normal = corr_gtex_normal.sort_values(ascending=False)[:top_genes]\n",
    "\n",
    "    high_corr_all_3 = highly_corr_BRCA.index.intersection(also_tcga_normal.index).intersection(also_gtex_normal.index)\n",
    "    \n",
    "    #changing the tcga normal and gtex variables, after calculating high_corr_all_3 bc\n",
    "    also_tcga_normal = also_tcga_normal[highly_corr_BRCA.index.intersection(also_tcga_normal.index)]\n",
    "    also_gtex_normal = also_gtex_normal[highly_corr_BRCA.index.intersection(also_gtex_normal.index)]    \n",
    "    \n",
    "    integrins.append(itg)\n",
    "    BRCA_high_corr_genes.append(highly_corr_BRCA.shape[0])\n",
    "    also_BRCA_normal.append(also_tcga_normal.shape[0])\n",
    "    also_breast_GTEx.append(also_gtex_normal.shape[0])   \n",
    "    highly_correlated_all_3.append(high_corr_all_3.shape[0])\n",
    "    print(itg, highly_corr_BRCA.shape[0], also_tcga_normal.shape[0], also_gtex_normal.shape[0],\n",
    "          high_corr_all_3.shape[0],sep='\\t')\n",
    "    \n",
    "tcga_primtum_gained_df=pd.DataFrame(data=list(zip(integrins,\n",
    "                                             BRCA_high_corr_genes,\n",
    "                                             also_BRCA_normal,\n",
    "                           also_breast_GTEx,\n",
    "                                                 highly_correlated_all_3)),\n",
    "             columns=['integrin',\n",
    "                      'BRCA_high_corr_genes_rank',\n",
    "                      'also_high_TCGA_normal',\n",
    "                      'also_high_GTEx',\n",
    "                     'high_corr_all_3']).sort_values(by='also_high_GTEx')\n",
    "del integrins, BRCA_high_corr_genes, also_BRCA_normal, also_breast_GTEx, highly_correlated_all_3\n",
    "tcga_primtum_gained_df    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "tcga_primtum_gained_df.to_excel(f'rankbased_BRCA_highly_corr_all_genes_top_{top_genes}.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "del gtex_corr_lost_df, tcga_primtum_gained_df, top_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
