{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.lines import Line2D\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "from os import system\n",
    "from IPython.display import clear_output\n",
    "from scipy.stats import pearsonr\n",
    "from scipy.stats import spearmanr\n",
    "import phik \n",
    "import seaborn as sns\n",
    "from phik import resources, report\n",
    "from phik.report import plot_correlation_matrix\n",
    "from itertools import combinations\n",
    "import random\n",
    "from tabulate import tabulate\n",
    "from IPython.display import display, HTML\n",
    "import graphviz\n",
    "import networkx as nx\n",
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "\n",
    "class add_path():\n",
    "    def __init__(self, path):\n",
    "        self.path = path\n",
    "\n",
    "    def __enter__(self):\n",
    "        sys.path.insert(0, self.path)\n",
    "\n",
    "    def __exit__(self, exc_type, exc_value, traceback):\n",
    "        try:\n",
    "            sys.path.remove(self.path)\n",
    "        except ValueError:\n",
    "            pass\n",
    "with add_path('/home/hshadman/.local/lib/python3.7/site-packages'):\n",
    "    matplotlib = __import__('matplotlib')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "matplotlib.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "from graphviz import Digraph\n",
    "gra = Digraph()\n",
    "gra.node('a', 'Machine Learning Errors')\n",
    "\n",
    "gra.node('b', 'RMSE')\n",
    "\n",
    "gra.node('c', 'MAE')\n",
    "gra.edges(['ab', 'ac'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gra"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(gra.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gra = Digraph()\n",
    "with gra.subgraph() as i:\n",
    "\n",
    "    i.attr(rank='same')\n",
    "\n",
    "    i.node('A')\n",
    "\n",
    "    i.node('X')\n",
    "\n",
    "gra.node('C')\n",
    "gra.edges(['AX'])\n",
    "gra\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import graphviz\n",
    "g = graphviz.Graph()\n",
    "g.node('A')\n",
    "g.node('B')\n",
    "g.edge('A','B')\n",
    "g"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# load functions \n",
    "# load the datasets, set cutoff values, correlation method (these are done individually so the system saves them globallly)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_toil_dataset = '../gtex_RSEM_gene_tpm.gz'\n",
    "correlation_method ='pearson'\n",
    "correlation_cutoff = 0.6\n",
    "tcga_hiseq_dataset = '/home/hshadman/integrins_expression/saghar_TCGA_Breast/TCGA_BRCA_HiSeqV2' \n",
    "tcga_toil_pancanc = \"../gtex_and_tcga/tcga_RSEM_gene_tpm.gz\"\n",
    "tcga_hiseq_corresponding_sample_info = '/home/hshadman/integrins_expression/saghar_TCGA_Breast/TCGA.BRCA.sampleMap_BRCA_clinicalMatrix'\n",
    "tcga_toil_sampinfo = \"../gtex_and_tcga/TCGA_phenotype_denseDataOnlyDownload.tsv.gz\"\n",
    "tcga_toil_clincinfo = \"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\"\n",
    "opposing_corr_cutoff = 0.1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#functions for data processing\n",
    "def gtex_data_processing(gtex_toil_dataset):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']\n",
    "    gtex_breast_data_all = pd.read_csv(gtex_toil_dataset,sep='\\t')\n",
    "    gtex_breast_data_all=gtex_breast_data_all.set_index('sample').copy()\n",
    "\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    gtex_breast_data_all.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    gtex_breast_data_all=gtex_breast_data_all.loc[gtex_breast_data_all.index.isin(list(gene_map.id.values))]\n",
    "    gtex_breast_data_all=pd.concat([gtex_breast_data_all,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:gtex_breast_data_all.shape[1]]\n",
    "    gtex_breast_data_all=gtex_breast_data_all.T.astype('float').copy()\n",
    "    df_int = gtex_breast_data_all.copy()\n",
    "    #drop unwanted integrins    \n",
    "    df_int=df_int.drop([undesired_integrin for undesired_integrin in df_int.columns if undesired_integrin in unwanted_integrins],axis=1)\n",
    "    #drop duplicate gene names (if any) \n",
    "    df_int=df_int.loc[:,~df_int.columns.duplicated()]    \n",
    "    \n",
    "    gtex_toil_phenotype = pd.read_csv('../GTEX_phenotype.gz',sep='\\t')\n",
    "    #subset for breast\n",
    "    gtex_toil_phenotype = gtex_toil_phenotype[gtex_toil_phenotype['body_site_detail (SMTSD)']=='Breast - Mammary Tissue']\n",
    "    #subset for female gender\n",
    "    gtex_toil_phenotype = gtex_toil_phenotype[gtex_toil_phenotype._gender=='female']\n",
    "\n",
    "    #select breast female gtex gene expression data\n",
    "    df_int=df_int.loc[df_int.index.isin(gtex_toil_phenotype.Sample.values)].copy()\n",
    "    \n",
    "    return df_int\n",
    "\n",
    "def tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    global tcga_brca_df, tcga_brca_sample_info\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]\n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop metastatic from main sample type data frame\n",
    "    #tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Metastatic')]\n",
    " \n",
    "    #separate normal and primtum (remember primary tumor contains multiple stages)\n",
    "    tcga_brca_normal_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Solid Tissue Normal']\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Primary Tumor']\n",
    "\n",
    "    #drop index of samples not in tcga_brca_df\n",
    "    tcga_brca_normal_patients = tcga_brca_normal_patients.drop(non_match for non_match in tcga_brca_normal_patients.index.values if non_match not in tcga_brca_df.columns)\n",
    "    tcga_brca_primtum_patients = tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "    #make sure both normal and tumor patients match, and drop patients not in tcga_brca_df\n",
    "    matched_patient_index_list=[]\n",
    "    for patient in tcga_brca_primtum_patients._PATIENT.values:\n",
    "        if patient in tcga_brca_normal_patients._PATIENT.values and patient in ['-'.join(patient_id.split('-')[0:3]) for patient_id in tcga_brca_df.columns]:\n",
    "            matched_patient_index_list.append(tcga_brca_sample_info[tcga_brca_sample_info._PATIENT==patient].index.values)\n",
    "    matched_patient_list=tcga_brca_sample_info.loc[[x for l in matched_patient_index_list for x in l]].copy()\n",
    "\n",
    "\n",
    "    # matched_patient_id_list=pd.Index(tcga_brca_normal_patients['_PATIENT']).intersection(pd.Index(tcga_brca_primtum_patients['_PATIENT']))\n",
    "    # matched_patient_index_list=tcga_brca_sample_info[tcga_brca_sample_info._PATIENT.isin(matched_patient_id_list)].index.values\n",
    "    # matched_patient_list = tcga_brca_sample_info.loc[[l for l in matched_patient_index_list]].copy()\n",
    "    # #ensure samples are present in tcga_brca_df\n",
    "    # matched_patient_list=matched_patient_list.loc[matched_patient_list.index.intersection(pd.Index(tcga_brca_df.columns))]\n",
    "\n",
    "    #separate normal and primary tumor\n",
    "    normal_matched_patient_list=matched_patient_list[matched_patient_list.sample_type=='Solid Tissue Normal']\n",
    "    primtum_matched_patient_list = matched_patient_list[matched_patient_list.sample_type=='Primary Tumor']\n",
    "\n",
    "    #matched normal and primary tumor samples\n",
    "    tcga_brca_df_normal_matched=tcga_brca_df[tcga_brca_df.columns[tcga_brca_df.columns.isin(normal_matched_patient_list.index.values)]]\n",
    "    tcga_brca_df_primtum_matched=tcga_brca_df[tcga_brca_df.columns[tcga_brca_df.columns.isin(primtum_matched_patient_list.index.values)]]\n",
    "\n",
    "    #test to see if normal and primary tumor patient IDs are a match\n",
    "    for i in [patient_id.split('-')[2] for patient_id in tcga_brca_df_normal_matched.columns]:\n",
    "        if i not in [patient_id.split('-')[2] for patient_id in tcga_brca_df_primtum_matched.columns]:\n",
    "            print('trouble')\n",
    "\n",
    "    #TCGA brca normal ALL\n",
    "    tcga_brca_df_normal_all = tcga_brca_df.T.loc[tcga_brca_normal_patients.index.values].copy()\n",
    "    # dataframe with all TCGA primary tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_all= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_normal_matched':tcga_brca_df_normal_matched,\n",
    "           'tcga_brca_df_primtum_matched':tcga_brca_df_primtum_matched,\n",
    "           'tcga_brca_df_normal_all':tcga_brca_df_normal_all,\n",
    "           'tcga_brca_df_primtum_all':tcga_brca_df_primtum_all}\n",
    "\n",
    "def correlation_selected_gene(provided_dataset,interest_gene,correlation_method):\n",
    "    dataset_corr = provided_dataset.corrwith(provided_dataset[interest_gene],\n",
    "                                             method=correlation_method).drop(interest_gene)\n",
    "    dataset_corr = dataset_corr.dropna(axis=0,how='any')\n",
    "    idx = dataset_corr.index.drop_duplicates(keep=False)\n",
    "    dataset_corr = dataset_corr.loc[idx]\n",
    "    return dataset_corr\n",
    "def pairwise_pearson_correlation(provided_dataset):\n",
    "    #columns have to be variables and rows have to be observations\n",
    "    pairwise_corr=pd.DataFrame(np.corrcoef(provided_dataset,rowvar=False),\n",
    "                               columns=provided_dataset.columns,index=provided_dataset.columns)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def early_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]\n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop metastatic from main sample type data frame\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Metastatic')]\n",
    "    #drop normal samples\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Solid Tissue Normal')]    \n",
    "\n",
    "    #make sure only primary tumor sample types\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info[tcga_brca_sample_info.sample_type=='Primary Tumor']\n",
    "    \n",
    "    #stage I and II only\n",
    "    #early_stages=['Stage IIB', 'Stage IA','Stage IIA','Stage I','Stage IB','Stage II']    \n",
    "    \n",
    "    #stage I only\n",
    "    early_stages=[ 'Stage IA','Stage I','Stage IB']    \n",
    "    \n",
    "    tcga_brca_primtum_patients=tcga_brca_primtum_patients[tcga_brca_primtum_patients.ajcc_pathologic_tumor_stage.isin(early_stages)]\n",
    "    \n",
    "    #drop index of samples not in tcga_brca_df\n",
    "\n",
    "    tcga_brca_primtum_patients =tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "\n",
    "    # dataframe with all TCGA primary early stage tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_early_stage= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_primtum_early_stage':tcga_brca_df_primtum_early_stage}\n",
    "\n",
    "def late_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo):\n",
    "    unwanted_integrins = ['ITGB3BP','ITGB1BP1','ITGB5-AS1','ITGA9-AS1','ITGB1P1','ITGB2-AS1', 'ITGB1BP2','ITGB1BP3']    \n",
    "    #First obtaining TCGA BRCA df from TCGA pan cancer\n",
    "    #sample info first file \n",
    "    pan_tcga_phenotype_sampinfo = pd.read_csv(tcga_toil_sampinfo,\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.set_index('sample')\n",
    "    #subset breast invasive carcinoma\n",
    "    chosen_canc='breast invasive carcinoma'\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo[pan_tcga_phenotype_sampinfo._primary_disease==chosen_canc]\n",
    "\n",
    "\n",
    "    #sample info second file\n",
    "    pan_tcga_phenotype_clinfo = pd.read_csv(\"../gtex_and_tcga/Survival_SupplementalTable_S1_20171025_xena_sp\",\n",
    "                                       sep='\\t',encoding='latin-1')\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.set_index('sample')\n",
    "    #subset for BRCA\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo[pan_tcga_phenotype_clinfo['cancer type abbreviation']=='BRCA']\n",
    "\n",
    "    #set common indices for both files\n",
    "    common_indices = pan_tcga_phenotype_sampinfo.index.intersection(pan_tcga_phenotype_clinfo.index)\n",
    "    pan_tcga_phenotype_sampinfo = pan_tcga_phenotype_sampinfo.loc[common_indices]\n",
    "    pan_tcga_phenotype_clinfo = pan_tcga_phenotype_clinfo.loc[common_indices]\n",
    "    tcga_brca_sample_info = pd.concat([pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo],axis=1)\n",
    "\n",
    "    #subset females only\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[tcga_brca_sample_info.gender =='FEMALE']\n",
    "\n",
    "\n",
    "    del pan_tcga_phenotype_sampinfo,pan_tcga_phenotype_clinfo\n",
    "\n",
    "    #retrieve TCGA pancan TOIL data - all cancers\n",
    "    pan_tcga = pd.read_csv(tcga_toil_pancanc,sep='\\t')\n",
    "\n",
    "    #subset the relevant TCGA BRCA data from pan cancer \n",
    "    chosen_samples = list(tcga_brca_sample_info.index.values)\n",
    "    tcga_brca_df = pan_tcga[pan_tcga.columns[pan_tcga.columns.isin(['sample']+chosen_samples)]]\n",
    "    tcga_brca_df = tcga_brca_df.set_index('sample')\n",
    "    del pan_tcga, chosen_samples\n",
    "    #next map gene names\n",
    "    gene_map = pd.read_csv(\"../gtex_and_tcga/probeMap_gencode.v23.annotation.gene.probemap\",sep='\\t')\n",
    "    #make another column with the isoform label stripped off\n",
    "    gene_map['id_neutral']=[geneid.split('.')[0] for geneid in gene_map.id.values]\n",
    "    protein_coding_genes=pd.read_csv(\"protein_coding_genes_mart_export.txt\",sep='\\t')\n",
    "    #remove rows with nan values\n",
    "    protein_coding_genes = protein_coding_genes.dropna(axis=0,how='any')\n",
    "    #subset probemap with protein coding genes\n",
    "    gene_map = gene_map[gene_map.id_neutral.isin([geneID.split('.')[0] for geneID in protein_coding_genes['Gene stable ID'].values])]\n",
    "\n",
    "    gene_map = gene_map[~gene_map.index.duplicated(keep='first')]\n",
    "    gene_id = gene_map.id.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('before dropping duplicates',len(gene_id),len(gene_name))\n",
    "    gene_duplicate = gene_map[gene_map.duplicated(subset=\"gene\")]\n",
    "    gene_duplicate = gene_duplicate.sort_values('gene', axis='index')\n",
    "    gene_map.drop_duplicates(subset=\"gene\", inplace=True, keep='first')\n",
    "    gene_id = gene_map.index.values.tolist()\n",
    "    gene_name = gene_map[\"gene\"].values.tolist()\n",
    "    print('after dropping duplicates',len(gene_id),len(gene_name))\n",
    "    # Drop all those duplicated genes from gene expression DataFrame\n",
    "    duplicate_drop = gene_duplicate.id.values  #these are the gene ids that are duplicated\n",
    "    tcga_brca_df.drop(duplicate_drop, axis='index',inplace=True)\n",
    "    gene_map_dict = dict(zip(gene_map.id, gene_map.gene))\n",
    "    tcga_brca_df=tcga_brca_df.loc[tcga_brca_df.index.isin(list(gene_map.id.values))]    \n",
    "    tcga_brca_df=pd.concat([tcga_brca_df,gene_map.set_index('id')],axis=1).set_index('gene').iloc[:,0:tcga_brca_df.shape[1]]\n",
    "    #now first step down, we have tcga_brca_df, and tcga_brca_sample_info\n",
    "    #subsequent steps derive from these two dataframe files\n",
    "    \n",
    "    \n",
    "    #drop unwanted integrins    \n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.isin(unwanted_integrins)]\n",
    "    #drop duplicate gene names (if any)\n",
    "    tcga_brca_df=tcga_brca_df.loc[~tcga_brca_df.index.duplicated()]    \n",
    "\n",
    "    #drop normal samples\n",
    "    tcga_brca_sample_info = tcga_brca_sample_info[~(tcga_brca_sample_info.sample_type=='Solid Tissue Normal')]    \n",
    "    #make sure only primary tumor sample types\n",
    "    tcga_brca_primtum_patients = tcga_brca_sample_info\n",
    "\n",
    "    late_stages=['Stage IIIA', 'Stage IV','Stage IIIC','Stage IIIB','Stage III']    \n",
    "    \n",
    "    tcga_brca_primtum_patients=tcga_brca_primtum_patients[tcga_brca_primtum_patients.ajcc_pathologic_tumor_stage.isin(late_stages)]\n",
    "    \n",
    "    #drop index of samples not in tcga_brca_df\n",
    "\n",
    "    tcga_brca_primtum_patients =tcga_brca_primtum_patients.drop(non_match for non_match in tcga_brca_primtum_patients.index.values if non_match not in tcga_brca_df.columns) \n",
    "\n",
    "\n",
    "\n",
    "    # dataframe with all TCGA primary late stage tumor samples (not necessarily matched)\n",
    "    tcga_brca_df_primtum_late_stage= tcga_brca_df.T.loc[tcga_brca_primtum_patients.index.values].copy()\n",
    "    \n",
    "    return {'tcga_brca_df_primtum_late_stage':tcga_brca_df_primtum_late_stage}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#files loading\n",
    "df_int=gtex_data_processing(gtex_toil_dataset)\n",
    "tcga_data=tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)\n",
    "tcga_brca_df_normal_all=tcga_data['tcga_brca_df_normal_all']\n",
    "tcga_brca_df_primtum_all=tcga_data['tcga_brca_df_primtum_all']\n",
    "tcga_brca_df_normal_matched=tcga_data['tcga_brca_df_normal_matched']\n",
    "tcga_brca_df_primtum_matched=tcga_data['tcga_brca_df_primtum_matched']\n",
    "tcga_brca_df_primtum_early_stage=early_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)['tcga_brca_df_primtum_early_stage']\n",
    "tcga_brca_df_primtum_late_stage=late_stage_tcga_data_processing(tcga_toil_pancanc,tcga_toil_sampinfo)['tcga_brca_df_primtum_late_stage']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
